{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import re\n",
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Scrap Team Lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_wiki_url(html, year, soup):\n",
    "    # Create a regular expression to match the target URLs.\n",
    "    url_regex = rf'\"http:\\/\\/{year}(.*)\\.igem\\.org\\/(.*)\"'\n",
    "    # Find all matching URLs in the HTML.\n",
    "    urls = re.findall(url_regex, html)\n",
    "    if len(urls) == 0:\n",
    "        # Create a regular expression to match the target URLs.\n",
    "        url_regex = rf'\"https:\\/\\/{year}(.*)\\.igem\\.org\\/(.*)\"'\n",
    "        # Find all matching URLs in the HTML.\n",
    "        urls = re.findall(url_regex, html)\n",
    "        if len(urls) > 0:\n",
    "            wiki = f\"https://{year}{urls[0][0]}.igem.org/{urls[0][1]}\"\n",
    "            wiki = wiki.split('\">Wiki<')[0]\n",
    "        elif len(urls) == 0:\n",
    "            # Create a regular expression to match the target URLs.\n",
    "            url_regex = rf'\"https:\\/\\/{year}\\.igem\\.wiki\\/(.*)\"'\n",
    "            # Find all matching URLs in the HTML.\n",
    "            urls = re.findall(url_regex, html)\n",
    "            if len(urls) > 0:\n",
    "                wiki = f\"https://{year}.igem.wiki/{urls[0]}\"\n",
    "                wiki = wiki.split('\">Wiki<')[0]\n",
    "            else: \n",
    "                wiki = None\n",
    "    else:\n",
    "        wiki = f\"https://{year}{urls[0][0]}.igem.org/{urls[0][1]}\"\n",
    "        wiki = wiki.split('\">Wiki<')[0]\n",
    "    # if soup.find('a', href=re.compile('wiki')) is not None:\n",
    "    #     wiki = soup.find('a', href=re.compile('wiki')).get('href')\n",
    "    # elif soup.find('a', href=re.compile('org')) is not None:\n",
    "    #     wiki = soup.find('a', href=re.compile('igem')).get('href')\n",
    "    # else:\n",
    "    #     wiki = None\n",
    "    return(wiki)\n",
    "def extract_each_row(row, year):\n",
    "    # Create a BeautifulSoup object\n",
    "    soup = BeautifulSoup(row, 'html.parser')\n",
    "\n",
    "    # Extract the team name\n",
    "    team_name = soup.find('a', class_='team_name').get_text()\n",
    "    link_team = soup.find('a', class_='team_name').get('href')\n",
    "    # wiki_url = soup.find('a', href=True).get('href')\n",
    "    wiki_url = get_wiki_url(str(row), year, soup)\n",
    "\n",
    "    # extract other information\n",
    "    infos = []\n",
    "    for a in row.split('<td data-order='):\n",
    "        infos.append(a.split('>')[-1])\n",
    "    region = infos[2]\n",
    "    country = infos[3]\n",
    "    track = infos[4]\n",
    "    kind = infos[5]\n",
    "    section = infos[6]\n",
    "    application = infos[7]\n",
    "    return(team_name, link_team, wiki_url, region, country, track, kind, section, application)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2008\n",
      "2009\n",
      "2010\n",
      "2011\n",
      "2012\n",
      "2013\n",
      "2014\n",
      "2015\n",
      "2016\n",
      "2017\n",
      "2018\n",
      "2019\n",
      "2020\n",
      "2021\n",
      "2022\n",
      "2023\n"
     ]
    }
   ],
   "source": [
    "teams = []\n",
    "links = []\n",
    "wikis = []\n",
    "regions = []\n",
    "countries = []\n",
    "tracks = []\n",
    "kinds = []\n",
    "sections = []\n",
    "apps = []\n",
    "years = []\n",
    "year_range = [2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023]\n",
    "\n",
    "\n",
    "for year in year_range:\n",
    "    url = 'https://old.igem.org/Team_List?year=' + str(year)\n",
    "    r = requests.get(url)\n",
    "    soup = BeautifulSoup(r.content, 'html.parser')\n",
    "    all = re.split(r\"<tr>\", str(soup.table.tbody.tr))[1:]\n",
    "    print(year)\n",
    "    for i in range(len(all)):\n",
    "        team_name, link_team, wiki_url, region, country, track, kind, section, application = extract_each_row(all[i], year)\n",
    "        teams.append(team_name)\n",
    "        links.append(link_team)\n",
    "        wikis.append(wiki_url)\n",
    "        regions.append(region)\n",
    "        countries.append(country)\n",
    "        tracks.append(track)\n",
    "        kinds.append(kind)\n",
    "        sections.append(section)\n",
    "        apps.append(application)\n",
    "        years.append(year)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Team name</th>\n",
       "      <th>Profile team</th>\n",
       "      <th>Wiki URL</th>\n",
       "      <th>Region</th>\n",
       "      <th>Country</th>\n",
       "      <th>Track</th>\n",
       "      <th>Kind</th>\n",
       "      <th>Section</th>\n",
       "      <th>Application</th>\n",
       "      <th>Year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Alberta_NINT</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=102</td>\n",
       "      <td>https://2008.igem.org/Team:Alberta_NINT</td>\n",
       "      <td>North America</td>\n",
       "      <td>Canada</td>\n",
       "      <td></td>\n",
       "      <td>Collegiate</td>\n",
       "      <td>iGEM</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bay_Area_RSI</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=80</td>\n",
       "      <td>https://2008.igem.org/Team:Bay_Area_RSI</td>\n",
       "      <td>North America</td>\n",
       "      <td>United States</td>\n",
       "      <td></td>\n",
       "      <td>Collegiate</td>\n",
       "      <td>iGEM</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BCCS-Bristol</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=135</td>\n",
       "      <td>https://2008.igem.org/Team:BCCS-Bristol</td>\n",
       "      <td>Europe</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td></td>\n",
       "      <td>Collegiate</td>\n",
       "      <td>iGEM</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Beijing_Normal</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=78</td>\n",
       "      <td>https://2008.igem.org/Team:Beijing_Normal</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "      <td></td>\n",
       "      <td>Collegiate</td>\n",
       "      <td>iGEM</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Bologna</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=79</td>\n",
       "      <td>https://2008.igem.org/Team:Bologna</td>\n",
       "      <td>Europe</td>\n",
       "      <td>Italy</td>\n",
       "      <td></td>\n",
       "      <td>Collegiate</td>\n",
       "      <td>iGEM</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4227</th>\n",
       "      <td>Zhejiang-United</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=4876</td>\n",
       "      <td>https://2023.igem.wiki/zhejiang-united</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "      <td>High School</td>\n",
       "      <td>Commercial</td>\n",
       "      <td>hs</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4228</th>\n",
       "      <td>ZJFH-Nanjing</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=4941</td>\n",
       "      <td>https://2023.igem.wiki/zjfh-nanjing</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "      <td>High School</td>\n",
       "      <td>Commercial</td>\n",
       "      <td>hs</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4229</th>\n",
       "      <td>ZJU-China</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=4628</td>\n",
       "      <td>https://2023.igem.wiki/zju-china</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "      <td>Agriculture</td>\n",
       "      <td>Collegiate</td>\n",
       "      <td>under</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4230</th>\n",
       "      <td>ZJUintl-China</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=4776</td>\n",
       "      <td>https://2023.igem.wiki/zjuintl-china</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "      <td>Therapeutics</td>\n",
       "      <td>Collegiate</td>\n",
       "      <td>under</td>\n",
       "      <td>Accepted</td>\n",
       "      <td>2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4231</th>\n",
       "      <td>ZJUT-China</td>\n",
       "      <td>https://old.igem.org/Team.cgi?team_id=4591</td>\n",
       "      <td>https://2023.igem.wiki/zjut-china</td>\n",
       "      <td>Asia</td>\n",
       "      <td>China</td>\n",
       "      <td>Bioremediation</td>\n",
       "      <td>Collegiate</td>\n",
       "      <td>under</td>\n",
       "      <td></td>\n",
       "      <td>2023</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4232 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Team name                                Profile team  \\\n",
       "0        Alberta_NINT   https://old.igem.org/Team.cgi?team_id=102   \n",
       "1        Bay_Area_RSI    https://old.igem.org/Team.cgi?team_id=80   \n",
       "2        BCCS-Bristol   https://old.igem.org/Team.cgi?team_id=135   \n",
       "3      Beijing_Normal    https://old.igem.org/Team.cgi?team_id=78   \n",
       "4             Bologna    https://old.igem.org/Team.cgi?team_id=79   \n",
       "...               ...                                         ...   \n",
       "4227  Zhejiang-United  https://old.igem.org/Team.cgi?team_id=4876   \n",
       "4228     ZJFH-Nanjing  https://old.igem.org/Team.cgi?team_id=4941   \n",
       "4229        ZJU-China  https://old.igem.org/Team.cgi?team_id=4628   \n",
       "4230    ZJUintl-China  https://old.igem.org/Team.cgi?team_id=4776   \n",
       "4231       ZJUT-China  https://old.igem.org/Team.cgi?team_id=4591   \n",
       "\n",
       "                                       Wiki URL         Region  \\\n",
       "0       https://2008.igem.org/Team:Alberta_NINT  North America   \n",
       "1       https://2008.igem.org/Team:Bay_Area_RSI  North America   \n",
       "2       https://2008.igem.org/Team:BCCS-Bristol         Europe   \n",
       "3     https://2008.igem.org/Team:Beijing_Normal           Asia   \n",
       "4            https://2008.igem.org/Team:Bologna         Europe   \n",
       "...                                         ...            ...   \n",
       "4227     https://2023.igem.wiki/zhejiang-united           Asia   \n",
       "4228        https://2023.igem.wiki/zjfh-nanjing           Asia   \n",
       "4229           https://2023.igem.wiki/zju-china           Asia   \n",
       "4230       https://2023.igem.wiki/zjuintl-china           Asia   \n",
       "4231          https://2023.igem.wiki/zjut-china           Asia   \n",
       "\n",
       "             Country           Track        Kind Section Application  Year  \n",
       "0             Canada                  Collegiate    iGEM    Accepted  2008  \n",
       "1      United States                  Collegiate    iGEM    Accepted  2008  \n",
       "2     United Kingdom                  Collegiate    iGEM    Accepted  2008  \n",
       "3              China                  Collegiate    iGEM    Accepted  2008  \n",
       "4              Italy                  Collegiate    iGEM    Accepted  2008  \n",
       "...              ...             ...         ...     ...         ...   ...  \n",
       "4227           China     High School  Commercial      hs    Accepted  2023  \n",
       "4228           China     High School  Commercial      hs    Accepted  2023  \n",
       "4229           China     Agriculture  Collegiate   under    Accepted  2023  \n",
       "4230           China    Therapeutics  Collegiate   under    Accepted  2023  \n",
       "4231           China  Bioremediation  Collegiate   under              2023  \n",
       "\n",
       "[4232 rows x 10 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create a DataFrame\n",
    "df = pd.DataFrame({\n",
    "    'Team name': teams,\n",
    "    'Profile team': links,\n",
    "    'Wiki URL': wikis,\n",
    "    'Region': regions,\n",
    "    'Country': countries,\n",
    "    'Track': tracks,\n",
    "    'Kind': kinds,\n",
    "    'Section': sections,\n",
    "    'Application': apps,\n",
    "    'Year' : years\n",
    "})\n",
    "\n",
    "\n",
    "# Print the DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 4232 entries, 0 to 4231\n",
      "Data columns (total 10 columns):\n",
      " #   Column        Non-Null Count  Dtype \n",
      "---  ------        --------------  ----- \n",
      " 0   Team name     4232 non-null   object\n",
      " 1   Profile team  4232 non-null   object\n",
      " 2   Wiki URL      4232 non-null   object\n",
      " 3   Region        4232 non-null   object\n",
      " 4   Country       4232 non-null   object\n",
      " 5   Track         4232 non-null   object\n",
      " 6   Kind          4232 non-null   object\n",
      " 7   Section       4232 non-null   object\n",
      " 8   Application   4232 non-null   object\n",
      " 9   Year          4232 non-null   int64 \n",
      "dtypes: int64(1), object(9)\n",
      "memory usage: 330.8+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the selected rows\n",
    "len(df[df.isnull().any(axis=1)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('team_lists_all_years.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Scrap Team Profile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://old.igem.org/Team.cgi?team_id=102' #2008\n",
    "# url = 'https://old.igem.org/Team.cgi?team_id=385' #2010\n",
    "# url = 'https://old.igem.org/Team.cgi?team_id=1161' #2013\n",
    "# url = 'https://old.igem.org/Team.cgi?team_id=2815' #2018\n",
    "# url = 'https://old.igem.org/Team.cgi?team_id=3743' # 2021\n",
    "# url = 'https://old.igem.org/Team.cgi?team_id=4769' #2023"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_title_abstract(url):\n",
    "    r = requests.get(url)\n",
    "    soup = BeautifulSoup(r.content, 'html.parser')\n",
    "    content = soup.findAll(id = 'table_abstract')[0].tr.td.get_text().split('\\n')\n",
    "    title = content[0]\n",
    "    abstract = \" \".join(content[1:])\n",
    "    return(title, abstract)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('-- Not provided yet --', '-- No abstract provided yet -- ')"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_title_abstract(url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "ename": "ConnectionError",
     "evalue": "('Connection aborted.', ConnectionResetError(54, 'Connection reset by peer'))",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mConnectionResetError\u001b[0m                      Traceback (most recent call last)",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connectionpool.py:790\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    789\u001b[0m \u001b[39m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 790\u001b[0m response \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_make_request(\n\u001b[1;32m    791\u001b[0m     conn,\n\u001b[1;32m    792\u001b[0m     method,\n\u001b[1;32m    793\u001b[0m     url,\n\u001b[1;32m    794\u001b[0m     timeout\u001b[39m=\u001b[39;49mtimeout_obj,\n\u001b[1;32m    795\u001b[0m     body\u001b[39m=\u001b[39;49mbody,\n\u001b[1;32m    796\u001b[0m     headers\u001b[39m=\u001b[39;49mheaders,\n\u001b[1;32m    797\u001b[0m     chunked\u001b[39m=\u001b[39;49mchunked,\n\u001b[1;32m    798\u001b[0m     retries\u001b[39m=\u001b[39;49mretries,\n\u001b[1;32m    799\u001b[0m     response_conn\u001b[39m=\u001b[39;49mresponse_conn,\n\u001b[1;32m    800\u001b[0m     preload_content\u001b[39m=\u001b[39;49mpreload_content,\n\u001b[1;32m    801\u001b[0m     decode_content\u001b[39m=\u001b[39;49mdecode_content,\n\u001b[1;32m    802\u001b[0m     \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mresponse_kw,\n\u001b[1;32m    803\u001b[0m )\n\u001b[1;32m    805\u001b[0m \u001b[39m# Everything went great!\u001b[39;00m\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connectionpool.py:536\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    535\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 536\u001b[0m     response \u001b[39m=\u001b[39m conn\u001b[39m.\u001b[39;49mgetresponse()\n\u001b[1;32m    537\u001b[0m \u001b[39mexcept\u001b[39;00m (BaseSSLError, \u001b[39mOSError\u001b[39;00m) \u001b[39mas\u001b[39;00m e:\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connection.py:461\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    460\u001b[0m \u001b[39m# Get the response from http.client.HTTPConnection\u001b[39;00m\n\u001b[0;32m--> 461\u001b[0m httplib_response \u001b[39m=\u001b[39m \u001b[39msuper\u001b[39;49m()\u001b[39m.\u001b[39;49mgetresponse()\n\u001b[1;32m    463\u001b[0m \u001b[39mtry\u001b[39;00m:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:1349\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1348\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m-> 1349\u001b[0m     response\u001b[39m.\u001b[39;49mbegin()\n\u001b[1;32m   1350\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mConnectionError\u001b[39;00m:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:316\u001b[0m, in \u001b[0;36mHTTPResponse.begin\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    315\u001b[0m \u001b[39mwhile\u001b[39;00m \u001b[39mTrue\u001b[39;00m:\n\u001b[0;32m--> 316\u001b[0m     version, status, reason \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_read_status()\n\u001b[1;32m    317\u001b[0m     \u001b[39mif\u001b[39;00m status \u001b[39m!=\u001b[39m CONTINUE:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:277\u001b[0m, in \u001b[0;36mHTTPResponse._read_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    276\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_read_status\u001b[39m(\u001b[39mself\u001b[39m):\n\u001b[0;32m--> 277\u001b[0m     line \u001b[39m=\u001b[39m \u001b[39mstr\u001b[39m(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mfp\u001b[39m.\u001b[39;49mreadline(_MAXLINE \u001b[39m+\u001b[39;49m \u001b[39m1\u001b[39;49m), \u001b[39m\"\u001b[39m\u001b[39miso-8859-1\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m    278\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(line) \u001b[39m>\u001b[39m _MAXLINE:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/socket.py:704\u001b[0m, in \u001b[0;36mSocketIO.readinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    703\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 704\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_sock\u001b[39m.\u001b[39;49mrecv_into(b)\n\u001b[1;32m    705\u001b[0m \u001b[39mexcept\u001b[39;00m timeout:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/ssl.py:1241\u001b[0m, in \u001b[0;36mSSLSocket.recv_into\u001b[0;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[1;32m   1238\u001b[0m         \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[1;32m   1239\u001b[0m           \u001b[39m\"\u001b[39m\u001b[39mnon-zero flags not allowed in calls to recv_into() on \u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m\n\u001b[1;32m   1240\u001b[0m           \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m\u001b[39m__class__\u001b[39m)\n\u001b[0;32m-> 1241\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mread(nbytes, buffer)\n\u001b[1;32m   1242\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/ssl.py:1099\u001b[0m, in \u001b[0;36mSSLSocket.read\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m   1098\u001b[0m \u001b[39mif\u001b[39;00m buffer \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[0;32m-> 1099\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_sslobj\u001b[39m.\u001b[39;49mread(\u001b[39mlen\u001b[39;49m, buffer)\n\u001b[1;32m   1100\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "\u001b[0;31mConnectionResetError\u001b[0m: [Errno 54] Connection reset by peer",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mProtocolError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    485\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 486\u001b[0m     resp \u001b[39m=\u001b[39m conn\u001b[39m.\u001b[39;49murlopen(\n\u001b[1;32m    487\u001b[0m         method\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mmethod,\n\u001b[1;32m    488\u001b[0m         url\u001b[39m=\u001b[39;49murl,\n\u001b[1;32m    489\u001b[0m         body\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mbody,\n\u001b[1;32m    490\u001b[0m         headers\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mheaders,\n\u001b[1;32m    491\u001b[0m         redirect\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    492\u001b[0m         assert_same_host\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    493\u001b[0m         preload_content\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    494\u001b[0m         decode_content\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    495\u001b[0m         retries\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mmax_retries,\n\u001b[1;32m    496\u001b[0m         timeout\u001b[39m=\u001b[39;49mtimeout,\n\u001b[1;32m    497\u001b[0m         chunked\u001b[39m=\u001b[39;49mchunked,\n\u001b[1;32m    498\u001b[0m     )\n\u001b[1;32m    500\u001b[0m \u001b[39mexcept\u001b[39;00m (ProtocolError, \u001b[39mOSError\u001b[39;00m) \u001b[39mas\u001b[39;00m err:\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connectionpool.py:844\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    842\u001b[0m     new_e \u001b[39m=\u001b[39m ProtocolError(\u001b[39m\"\u001b[39m\u001b[39mConnection aborted.\u001b[39m\u001b[39m\"\u001b[39m, new_e)\n\u001b[0;32m--> 844\u001b[0m retries \u001b[39m=\u001b[39m retries\u001b[39m.\u001b[39;49mincrement(\n\u001b[1;32m    845\u001b[0m     method, url, error\u001b[39m=\u001b[39;49mnew_e, _pool\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m, _stacktrace\u001b[39m=\u001b[39;49msys\u001b[39m.\u001b[39;49mexc_info()[\u001b[39m2\u001b[39;49m]\n\u001b[1;32m    846\u001b[0m )\n\u001b[1;32m    847\u001b[0m retries\u001b[39m.\u001b[39msleep()\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/util/retry.py:470\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    469\u001b[0m \u001b[39mif\u001b[39;00m read \u001b[39mis\u001b[39;00m \u001b[39mFalse\u001b[39;00m \u001b[39mor\u001b[39;00m method \u001b[39mis\u001b[39;00m \u001b[39mNone\u001b[39;00m \u001b[39mor\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_is_method_retryable(method):\n\u001b[0;32m--> 470\u001b[0m     \u001b[39mraise\u001b[39;00m reraise(\u001b[39mtype\u001b[39;49m(error), error, _stacktrace)\n\u001b[1;32m    471\u001b[0m \u001b[39melif\u001b[39;00m read \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/util/util.py:38\u001b[0m, in \u001b[0;36mreraise\u001b[0;34m(tp, value, tb)\u001b[0m\n\u001b[1;32m     37\u001b[0m \u001b[39mif\u001b[39;00m value\u001b[39m.\u001b[39m__traceback__ \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m tb:\n\u001b[0;32m---> 38\u001b[0m     \u001b[39mraise\u001b[39;00m value\u001b[39m.\u001b[39mwith_traceback(tb)\n\u001b[1;32m     39\u001b[0m \u001b[39mraise\u001b[39;00m value\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connectionpool.py:790\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    789\u001b[0m \u001b[39m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 790\u001b[0m response \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_make_request(\n\u001b[1;32m    791\u001b[0m     conn,\n\u001b[1;32m    792\u001b[0m     method,\n\u001b[1;32m    793\u001b[0m     url,\n\u001b[1;32m    794\u001b[0m     timeout\u001b[39m=\u001b[39;49mtimeout_obj,\n\u001b[1;32m    795\u001b[0m     body\u001b[39m=\u001b[39;49mbody,\n\u001b[1;32m    796\u001b[0m     headers\u001b[39m=\u001b[39;49mheaders,\n\u001b[1;32m    797\u001b[0m     chunked\u001b[39m=\u001b[39;49mchunked,\n\u001b[1;32m    798\u001b[0m     retries\u001b[39m=\u001b[39;49mretries,\n\u001b[1;32m    799\u001b[0m     response_conn\u001b[39m=\u001b[39;49mresponse_conn,\n\u001b[1;32m    800\u001b[0m     preload_content\u001b[39m=\u001b[39;49mpreload_content,\n\u001b[1;32m    801\u001b[0m     decode_content\u001b[39m=\u001b[39;49mdecode_content,\n\u001b[1;32m    802\u001b[0m     \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mresponse_kw,\n\u001b[1;32m    803\u001b[0m )\n\u001b[1;32m    805\u001b[0m \u001b[39m# Everything went great!\u001b[39;00m\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connectionpool.py:536\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    535\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 536\u001b[0m     response \u001b[39m=\u001b[39m conn\u001b[39m.\u001b[39;49mgetresponse()\n\u001b[1;32m    537\u001b[0m \u001b[39mexcept\u001b[39;00m (BaseSSLError, \u001b[39mOSError\u001b[39;00m) \u001b[39mas\u001b[39;00m e:\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/urllib3/connection.py:461\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    460\u001b[0m \u001b[39m# Get the response from http.client.HTTPConnection\u001b[39;00m\n\u001b[0;32m--> 461\u001b[0m httplib_response \u001b[39m=\u001b[39m \u001b[39msuper\u001b[39;49m()\u001b[39m.\u001b[39;49mgetresponse()\n\u001b[1;32m    463\u001b[0m \u001b[39mtry\u001b[39;00m:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:1349\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1348\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m-> 1349\u001b[0m     response\u001b[39m.\u001b[39;49mbegin()\n\u001b[1;32m   1350\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mConnectionError\u001b[39;00m:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:316\u001b[0m, in \u001b[0;36mHTTPResponse.begin\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    315\u001b[0m \u001b[39mwhile\u001b[39;00m \u001b[39mTrue\u001b[39;00m:\n\u001b[0;32m--> 316\u001b[0m     version, status, reason \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_read_status()\n\u001b[1;32m    317\u001b[0m     \u001b[39mif\u001b[39;00m status \u001b[39m!=\u001b[39m CONTINUE:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/http/client.py:277\u001b[0m, in \u001b[0;36mHTTPResponse._read_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    276\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_read_status\u001b[39m(\u001b[39mself\u001b[39m):\n\u001b[0;32m--> 277\u001b[0m     line \u001b[39m=\u001b[39m \u001b[39mstr\u001b[39m(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mfp\u001b[39m.\u001b[39;49mreadline(_MAXLINE \u001b[39m+\u001b[39;49m \u001b[39m1\u001b[39;49m), \u001b[39m\"\u001b[39m\u001b[39miso-8859-1\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m    278\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(line) \u001b[39m>\u001b[39m _MAXLINE:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/socket.py:704\u001b[0m, in \u001b[0;36mSocketIO.readinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    703\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 704\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_sock\u001b[39m.\u001b[39;49mrecv_into(b)\n\u001b[1;32m    705\u001b[0m \u001b[39mexcept\u001b[39;00m timeout:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/ssl.py:1241\u001b[0m, in \u001b[0;36mSSLSocket.recv_into\u001b[0;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[1;32m   1238\u001b[0m         \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[1;32m   1239\u001b[0m           \u001b[39m\"\u001b[39m\u001b[39mnon-zero flags not allowed in calls to recv_into() on \u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m\n\u001b[1;32m   1240\u001b[0m           \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m\u001b[39m__class__\u001b[39m)\n\u001b[0;32m-> 1241\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mread(nbytes, buffer)\n\u001b[1;32m   1242\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "File \u001b[0;32m/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/ssl.py:1099\u001b[0m, in \u001b[0;36mSSLSocket.read\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m   1098\u001b[0m \u001b[39mif\u001b[39;00m buffer \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[0;32m-> 1099\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_sslobj\u001b[39m.\u001b[39;49mread(\u001b[39mlen\u001b[39;49m, buffer)\n\u001b[1;32m   1100\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "\u001b[0;31mProtocolError\u001b[0m: ('Connection aborted.', ConnectionResetError(54, 'Connection reset by peer'))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mConnectionError\u001b[0m                           Traceback (most recent call last)",
      "\u001b[1;32m/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb Cell 13\u001b[0m line \u001b[0;36m4\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=1'>2</a>\u001b[0m abstracts \u001b[39m=\u001b[39m []\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=2'>3</a>\u001b[0m \u001b[39mfor\u001b[39;00m i \u001b[39min\u001b[39;00m df[\u001b[39m'\u001b[39m\u001b[39mProfile team\u001b[39m\u001b[39m'\u001b[39m]:\n\u001b[0;32m----> <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=3'>4</a>\u001b[0m     title, \u001b[39mabs\u001b[39m \u001b[39m=\u001b[39m get_title_abstract(i)\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=4'>5</a>\u001b[0m     titles\u001b[39m.\u001b[39mappend(title)\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=5'>6</a>\u001b[0m     abstracts\u001b[39m.\u001b[39mappend(\u001b[39mabs\u001b[39m)\n",
      "\u001b[1;32m/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb Cell 13\u001b[0m line \u001b[0;36m2\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=0'>1</a>\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mget_title_abstract\u001b[39m(url):\n\u001b[0;32m----> <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=1'>2</a>\u001b[0m     r \u001b[39m=\u001b[39m requests\u001b[39m.\u001b[39;49mget(url)\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=2'>3</a>\u001b[0m     soup \u001b[39m=\u001b[39m BeautifulSoup(r\u001b[39m.\u001b[39mcontent, \u001b[39m'\u001b[39m\u001b[39mhtml.parser\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/macbookpro/Documents/GitHub/project-discovery-dashboard/scrap.ipynb#X43sZmlsZQ%3D%3D?line=3'>4</a>\u001b[0m     content \u001b[39m=\u001b[39m soup\u001b[39m.\u001b[39mfindAll(\u001b[39mid\u001b[39m \u001b[39m=\u001b[39m \u001b[39m'\u001b[39m\u001b[39mtable_abstract\u001b[39m\u001b[39m'\u001b[39m)[\u001b[39m0\u001b[39m]\u001b[39m.\u001b[39mtr\u001b[39m.\u001b[39mtd\u001b[39m.\u001b[39mget_text()\u001b[39m.\u001b[39msplit(\u001b[39m'\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m'\u001b[39m)\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/api.py:73\u001b[0m, in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mget\u001b[39m(url, params\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs):\n\u001b[1;32m     63\u001b[0m \u001b[39m    \u001b[39m\u001b[39mr\u001b[39m\u001b[39m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \n\u001b[1;32m     65\u001b[0m \u001b[39m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[39m    :rtype: requests.Response\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[39m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 73\u001b[0m     \u001b[39mreturn\u001b[39;00m request(\u001b[39m\"\u001b[39;49m\u001b[39mget\u001b[39;49m\u001b[39m\"\u001b[39;49m, url, params\u001b[39m=\u001b[39;49mparams, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[39m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[39m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[1;32m     57\u001b[0m \u001b[39m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[1;32m     58\u001b[0m \u001b[39mwith\u001b[39;00m sessions\u001b[39m.\u001b[39mSession() \u001b[39mas\u001b[39;00m session:\n\u001b[0;32m---> 59\u001b[0m     \u001b[39mreturn\u001b[39;00m session\u001b[39m.\u001b[39;49mrequest(method\u001b[39m=\u001b[39;49mmethod, url\u001b[39m=\u001b[39;49murl, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    584\u001b[0m send_kwargs \u001b[39m=\u001b[39m {\n\u001b[1;32m    585\u001b[0m     \u001b[39m\"\u001b[39m\u001b[39mtimeout\u001b[39m\u001b[39m\"\u001b[39m: timeout,\n\u001b[1;32m    586\u001b[0m     \u001b[39m\"\u001b[39m\u001b[39mallow_redirects\u001b[39m\u001b[39m\"\u001b[39m: allow_redirects,\n\u001b[1;32m    587\u001b[0m }\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[39m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49msend(prep, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49msend_kwargs)\n\u001b[1;32m    591\u001b[0m \u001b[39mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/sessions.py:725\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    722\u001b[0m \u001b[39mif\u001b[39;00m allow_redirects:\n\u001b[1;32m    723\u001b[0m     \u001b[39m# Redirect resolving generator.\u001b[39;00m\n\u001b[1;32m    724\u001b[0m     gen \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mresolve_redirects(r, request, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n\u001b[0;32m--> 725\u001b[0m     history \u001b[39m=\u001b[39m [resp \u001b[39mfor\u001b[39;00m resp \u001b[39min\u001b[39;00m gen]\n\u001b[1;32m    726\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m    727\u001b[0m     history \u001b[39m=\u001b[39m []\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/sessions.py:725\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    722\u001b[0m \u001b[39mif\u001b[39;00m allow_redirects:\n\u001b[1;32m    723\u001b[0m     \u001b[39m# Redirect resolving generator.\u001b[39;00m\n\u001b[1;32m    724\u001b[0m     gen \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mresolve_redirects(r, request, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n\u001b[0;32m--> 725\u001b[0m     history \u001b[39m=\u001b[39m [resp \u001b[39mfor\u001b[39;00m resp \u001b[39min\u001b[39;00m gen]\n\u001b[1;32m    726\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m    727\u001b[0m     history \u001b[39m=\u001b[39m []\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/sessions.py:266\u001b[0m, in \u001b[0;36mSessionRedirectMixin.resolve_redirects\u001b[0;34m(self, resp, req, stream, timeout, verify, cert, proxies, yield_requests, **adapter_kwargs)\u001b[0m\n\u001b[1;32m    263\u001b[0m     \u001b[39myield\u001b[39;00m req\n\u001b[1;32m    264\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m--> 266\u001b[0m     resp \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49msend(\n\u001b[1;32m    267\u001b[0m         req,\n\u001b[1;32m    268\u001b[0m         stream\u001b[39m=\u001b[39;49mstream,\n\u001b[1;32m    269\u001b[0m         timeout\u001b[39m=\u001b[39;49mtimeout,\n\u001b[1;32m    270\u001b[0m         verify\u001b[39m=\u001b[39;49mverify,\n\u001b[1;32m    271\u001b[0m         cert\u001b[39m=\u001b[39;49mcert,\n\u001b[1;32m    272\u001b[0m         proxies\u001b[39m=\u001b[39;49mproxies,\n\u001b[1;32m    273\u001b[0m         allow_redirects\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    274\u001b[0m         \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49madapter_kwargs,\n\u001b[1;32m    275\u001b[0m     )\n\u001b[1;32m    277\u001b[0m     extract_cookies_to_jar(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mcookies, prepared_request, resp\u001b[39m.\u001b[39mraw)\n\u001b[1;32m    279\u001b[0m     \u001b[39m# extract redirect url, if any, for the next loop\u001b[39;00m\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    700\u001b[0m start \u001b[39m=\u001b[39m preferred_clock()\n\u001b[1;32m    702\u001b[0m \u001b[39m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[39m=\u001b[39m adapter\u001b[39m.\u001b[39;49msend(request, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n\u001b[1;32m    705\u001b[0m \u001b[39m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[1;32m    706\u001b[0m elapsed \u001b[39m=\u001b[39m preferred_clock() \u001b[39m-\u001b[39m start\n",
      "File \u001b[0;32m~/Library/Python/3.9/lib/python/site-packages/requests/adapters.py:501\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    486\u001b[0m     resp \u001b[39m=\u001b[39m conn\u001b[39m.\u001b[39murlopen(\n\u001b[1;32m    487\u001b[0m         method\u001b[39m=\u001b[39mrequest\u001b[39m.\u001b[39mmethod,\n\u001b[1;32m    488\u001b[0m         url\u001b[39m=\u001b[39murl,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    497\u001b[0m         chunked\u001b[39m=\u001b[39mchunked,\n\u001b[1;32m    498\u001b[0m     )\n\u001b[1;32m    500\u001b[0m \u001b[39mexcept\u001b[39;00m (ProtocolError, \u001b[39mOSError\u001b[39;00m) \u001b[39mas\u001b[39;00m err:\n\u001b[0;32m--> 501\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mConnectionError\u001b[39;00m(err, request\u001b[39m=\u001b[39mrequest)\n\u001b[1;32m    503\u001b[0m \u001b[39mexcept\u001b[39;00m MaxRetryError \u001b[39mas\u001b[39;00m e:\n\u001b[1;32m    504\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39misinstance\u001b[39m(e\u001b[39m.\u001b[39mreason, ConnectTimeoutError):\n\u001b[1;32m    505\u001b[0m         \u001b[39m# TODO: Remove this in 3.0.0: see #2811\u001b[39;00m\n",
      "\u001b[0;31mConnectionError\u001b[0m: ('Connection aborted.', ConnectionResetError(54, 'Connection reset by peer'))"
     ]
    }
   ],
   "source": [
    "titles = []\n",
    "abstracts = []\n",
    "for i in df['Profile team']:\n",
    "    title, abs = get_title_abstract(i)\n",
    "    titles.append(title)\n",
    "    abstracts.append(abs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Scrap Wiki"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://2022.igem.wiki/tu-eindhoven/engineering'\n",
    "r = requests.get(url)\n",
    "soup = BeautifulSoup(r.content, 'html.parser')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<!DOCTYPE html>\n",
      "<html lang=\"en\">\n",
      " <head>\n",
      "  <!-- Required meta tags -->\n",
      "  <meta charset=\"utf-8\"/>\n",
      "  <meta content=\"width=device-width, initial-scale=1\" name=\"viewport\"/>\n",
      "  <link href=\"https://static.igem.wiki/common/icons/favicons/igem-2022.svg\" rel=\"shortcut icon\"/>\n",
      "  <link href=\"https://creativecommons.org/licenses/by/4.0/\" rel=\"license\"/>\n",
      "  <!-- Bootstrap CSS -->\n",
      "  <link href=\"static/bootstrap.min.css\" rel=\"stylesheet\"/>\n",
      "  <!-- Custom CSS -->\n",
      "  <link href=\"static/style.css\" rel=\"stylesheet\"/>\n",
      "  <title>\n",
      "   Engineering Success | TU-Eindhoven - iGEM 2022\n",
      "  </title>\n",
      " </head>\n",
      " <body>\n",
      "  <!-- Navigation -->\n",
      "  <nav class=\"navbar navbar-expand-lg navbar-dark fixed-top\" style=\"background-color: #1b263b;\">\n",
      "   <div class=\"container\">\n",
      "    <!---- TEAM NAME ---->\n",
      "    <a class=\"navbar-brand\" href=\"https://2022.igem.wiki/tu-eindhoven/\" style=\"max-width: 79%\">\n",
      "     <img class=\"w-25\" src=\"https://static.igem.wiki/teams/4160/wiki/footer/mpact-wit.svg\" style=\"padding: 5px\"/>\n",
      "     TU-Eindhoven\n",
      "    </a>\n",
      "    <!---- SMALL SCREEN MENU ICON ---->\n",
      "    <button aria-controls=\"navbarSupportedContent\" aria-expanded=\"false\" aria-label=\"Toggle navigation\" class=\"navbar-toggler\" data-bs-target=\"#navbarSupportedContent\" data-bs-toggle=\"collapse\" type=\"button\">\n",
      "     <span class=\"navbar-toggler-icon\">\n",
      "     </span>\n",
      "    </button>\n",
      "    <div class=\"collapse navbar-collapse\" id=\"navbarSupportedContent\">\n",
      "     <ul class=\"navbar-nav ml-auto left-aligned\">\n",
      "      <!---- HOME ---->\n",
      "      <li class=\"nav-item\">\n",
      "       <a class=\"nav-link\" href=\"index.html\">\n",
      "        Home\n",
      "       </a>\n",
      "      </li>\n",
      "      <!---- TEAM ---->\n",
      "      <li class=\"nav-item dropdown\">\n",
      "       <a aria-expanded=\"false\" class=\"nav-link dropdown-toggle\" data-bs-toggle=\"dropdown\" href=\"team\" id=\"navbarDropdown\" role=\"button\">\n",
      "        Team\n",
      "       </a>\n",
      "       <ul aria-labelledby=\"navbarDropdown\" class=\"dropdown-menu\">\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"team\">\n",
      "          Our team\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"attributions\">\n",
      "          Attributions\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"sponsors\">\n",
      "          Sponsors\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"collaborations\">\n",
      "          Collaborations\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"partnership\">\n",
      "          Partnership\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"awards\">\n",
      "          Awards\n",
      "         </a>\n",
      "        </li>\n",
      "       </ul>\n",
      "      </li>\n",
      "      <!---- PROJECT ---->\n",
      "      <li class=\"nav-item dropdown\">\n",
      "       <a aria-expanded=\"false\" class=\"nav-link dropdown-toggle\" data-bs-toggle=\"dropdown\" href=\"#\" id=\"navbarDropdown\" role=\"button\">\n",
      "        Project\n",
      "       </a>\n",
      "       <ul aria-labelledby=\"navbarDropdown\" class=\"dropdown-menu\">\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"description\">\n",
      "          Description\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"engineering\">\n",
      "          Engineering\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"implementation\">\n",
      "          Implementation\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"proof-of-concept\">\n",
      "          Proof of Concept\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"entrepreneurship\">\n",
      "          Entrepreneurship\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"model\">\n",
      "          Model\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"medal\">\n",
      "          Medal criteria\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"https://old.igem.org/2022_Judging_Form?id=4160\" target=\"_blank\">\n",
      "          Judging form\n",
      "         </a>\n",
      "        </li>\n",
      "       </ul>\n",
      "      </li>\n",
      "      <!---- Lab ---->\n",
      "      <li class=\"nav-item dropdown\">\n",
      "       <a aria-expanded=\"false\" class=\"nav-link dropdown-toggle\" data-bs-toggle=\"dropdown\" href=\"#\" id=\"navbarDropdown\" role=\"button\">\n",
      "        Lab\n",
      "       </a>\n",
      "       <ul aria-labelledby=\"navbarDropdown\" class=\"dropdown-menu\">\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"results\">\n",
      "          Results\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"experiments\">\n",
      "          Experiments\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"notebook\">\n",
      "          Notebook\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"parts\">\n",
      "          Parts\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"part-collection\">\n",
      "          Part Collection\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"improve\">\n",
      "          Part improvement\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"contribution\">\n",
      "          Contribution\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"primers_gblocks\">\n",
      "          Primers and gBlocks\n",
      "         </a>\n",
      "        </li>\n",
      "       </ul>\n",
      "      </li>\n",
      "      <!---- Public outreach ---->\n",
      "      <li class=\"nav-item dropdown\">\n",
      "       <a aria-expanded=\"false\" class=\"nav-link dropdown-toggle\" data-bs-toggle=\"dropdown\" href=\"#\" id=\"navbarDropdown\" role=\"button\">\n",
      "        Outreach\n",
      "       </a>\n",
      "       <ul aria-labelledby=\"navbarDropdown\" class=\"dropdown-menu\">\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"communication\">\n",
      "          Communication\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a class=\"dropdown-item\" href=\"education\">\n",
      "          Education\n",
      "         </a>\n",
      "        </li>\n",
      "       </ul>\n",
      "      </li>\n",
      "      <!---- HUMAN PRACTICES ---->\n",
      "      <li class=\"nav-item\">\n",
      "       <a class=\"nav-link\" href=\"human-practices\">\n",
      "        Human Practices\n",
      "       </a>\n",
      "      </li>\n",
      "      <!---- Safety ---->\n",
      "      <li class=\"nav-item\">\n",
      "       <a class=\"nav-link\" href=\"safety\">\n",
      "        Safety\n",
      "       </a>\n",
      "      </li>\n",
      "     </ul>\n",
      "    </div>\n",
      "   </div>\n",
      "  </nav>\n",
      "  <!-- Header -->\n",
      "  <div class=\"headerdiv\">\n",
      "   <!--Due to difficulty with the header sometimes consisting out of two words, the styling will\n",
      "            be done in the actual html page-->\n",
      "  </div>\n",
      "  <div style=\"position:relative\">\n",
      "   <!-- Page Content-->\n",
      "   <div class=\"lead_div\">\n",
      "    <div class=\"center-title\" id=\"center-title\">\n",
      "     <h1 id=\"leadtitle\" style=\"color:white;\">\n",
      "      <span class=\"capital-green\">\n",
      "       E\n",
      "      </span>\n",
      "      ngineering\n",
      "      <br/>\n",
      "      <span class=\"capital-green\">\n",
      "       S\n",
      "      </span>\n",
      "      uccess\n",
      "     </h1>\n",
      "    </div>\n",
      "   </div>\n",
      "   <!-- Curved Lines header-->\n",
      "   <img class=\"dashedline\" src=\"https://static.igem.wiki/teams/4160/wiki/general/dashedline-green.svg\">\n",
      "    <img class=\"background-image-header\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/header/engineering-achtergrond-kleiner-01.png\"/>\n",
      "    <div class=\"cropped\">\n",
      "     <img class=\"backgroundcurves\" src=\"https://static.igem.wiki/teams/4160/wiki/general/upper-background.svg\"/>\n",
      "    </div>\n",
      "    <div class=\"blueblock\">\n",
      "     <!-- Quote-->\n",
      "     <div class=\"quote\" style=\"margin-top:0;\">\n",
      "      <p>\n",
      "       <i>\n",
      "        \"Some of the best lessons we ever learn we learn from our mistakes and failures. The error of the past is the wisdom and success of the future.\"\n",
      "       </i>\n",
      "       - Tryon Edwards\n",
      "      </p>\n",
      "     </div>\n",
      "    </div>\n",
      "    <div class=\"layout\">\n",
      "     <!-- Start actual content of the page -->\n",
      "     <div class=\"blueblock outline\">\n",
      "      <div>\n",
      "      </div>\n",
      "      <div class=\"container container-custom\">\n",
      "       <div class=\"row mb-5 gx-0\" id=\"introduction\">\n",
      "        <div class=\"col-1 d-flex justify-content-center\">\n",
      "         <div class=\"bar-green\" style=\"height: 100%;\">\n",
      "         </div>\n",
      "        </div>\n",
      "        <div class=\"col-10\">\n",
      "         <div class=\"row gx-5\">\n",
      "          <div class=\"col-12 mb-3\">\n",
      "           <p>\n",
      "            To design and build parts of our biological system, we used engineering principles. In addition, the engineering design cycle was used to tackle challenges in our project. This engineering design cycle consists of four stages Design, Build, Test, and Learn. Nevertheless, we complemented the engineering design cycle with a topic Research, since research is often necessary to successfully design a system. After reflection on the first iteration cycle, one usually starts a new iteration of the cycle from the research step onwards. However, it is possible to enter any stage in a new cycle after reflection. The engineering design cycle is visualized in\n",
      "            <b class=\"bold\">\n",
      "             Figure 1\n",
      "            </b>\n",
      "            .\n",
      "           </p>\n",
      "           <picture>\n",
      "            <source media=\"(min-width: 1000px)\" srcset=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/header/design-proces-header.svg\"/>\n",
      "            <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/header/design-proces-header.svg\" style=\"padding:3% 12% 0 12%; float:left;\"/>\n",
      "            <figcaption style=\"font-size:13px\">\n",
      "             <b>\n",
      "              Figure 1 | The engineering design cycle.\n",
      "             </b>\n",
      "             The engineering design cycle consists of the following stages: Research, Design, Build, Test, and Learn. After an iteration, one usually goes back to Research, however, it is in principle possible to go back to any stage; this is visualized with the blue dashed line.\n",
      "            </figcaption>\n",
      "           </picture>\n",
      "           <br/>\n",
      "           <p>\n",
      "            The iterations of the design cycle for two aspects of our project are described on this page:\n",
      "            <b class=\"bold\">\n",
      "             cloning strategies\n",
      "            </b>\n",
      "            and\n",
      "            <b class=\"bold\">\n",
      "             educational lessons for elementary schools\n",
      "            </b>\n",
      "            . These cycles show how we tackled challenges during our project, what we learned from each engineering iteration, and what we did to improve the design.\n",
      "           </p>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div>\n",
      "      </div>\n",
      "     </div>\n",
      "    </div>\n",
      "    <!-- lab and education tabs -->\n",
      "    <div class=\"blueblock\">\n",
      "     <div class=\"container text-center\">\n",
      "      <div class=\"row gy-5 d-flex justify-content-center\">\n",
      "       <div class=\"col-4 d-flex flex-row align-items-center hover-goals subtitle-selected\" id=\"lab_button\" onclick=\"openOtherDiv('lab')\">\n",
      "        <img class=\"w-25\" src=\"https://static.igem.wiki/teams/4160/wiki/model/miscellaneous/c-button-green.svg\"/>\n",
      "        <h1 class=\"goals-texts\">\n",
      "         Lab\n",
      "        </h1>\n",
      "       </div>\n",
      "       <div class=\"col-4 d-flex flex-row align-items-center hover-goals subtitle-unselected\" id=\"education_button\" onclick=\"openOtherDiv('education')\">\n",
      "        <img class=\"w-25\" src=\"https://static.igem.wiki/teams/4160/wiki/model/miscellaneous/c-button-green.svg\"/>\n",
      "        <h1 class=\"goals-texts\">\n",
      "         Education\n",
      "        </h1>\n",
      "       </div>\n",
      "      </div>\n",
      "     </div>\n",
      "    </div>\n",
      "    <div id=\"lab_div\">\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:28%; left:-5%; width:35%; z-index:-1\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:21%; left:57%; width:20%; -ms-transform: rotate(42deg) ; transform: rotate(42deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:25%; left:80%; width:28%; -ms-transform: rotate(180deg); transform: rotate(180deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:37%; left:25%; width:15%; -ms-transform: rotate(316deg); transform: rotate(316deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:44%; left:72%; width:30%; -ms-transform: rotate(160deg); transform: rotate(160deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:48%; left:10%; width:20%; -ms-transform: rotate(285deg); transform: rotate(285deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:56%; left:2%; width:30%; -ms-transform: rotate(200deg); transform: rotate(200deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:61%; left:68%; width:10%; -ms-transform: rotate(30deg); transform: rotate(30deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:65%; left:70%; width:35%; -ms-transform: rotate(155deg); transform: rotate(155deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:74%; left:4%; width:22%; -ms-transform: rotate(-45deg); transform: rotate(-45deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:86%; left:88%; width:10%; -ms-transform: rotate(25deg); transform: rotate(25deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:83%; left:65%; width:17%; -ms-transform: rotate(130deg); transform: rotate(130deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div class=\"layout\">\n",
      "      <!-- Start actual content of the page -->\n",
      "      <div class=\"blueblock outline\">\n",
      "       <div>\n",
      "       </div>\n",
      "       <div class=\"container container-custom\">\n",
      "        <div class=\"row mb-5 gx-0\" id=\"introduction\">\n",
      "         <div class=\"col-1 d-flex justify-content-center\">\n",
      "          <!--<div class=\"bar-green\" style=\"height: 100%;\"></div> -->\n",
      "         </div>\n",
      "         <div class=\"col-10\">\n",
      "          <div class=\"row gx-5\">\n",
      "           <div class=\"col-12 mb-3\">\n",
      "            <br/>\n",
      "            <br/>\n",
      "            <p>\n",
      "             On this\n",
      "             <b class=\"bold\">\n",
      "              engineering success\n",
      "             </b>\n",
      "             wiki page, we discuss multiple iterations of the DTBL cycle performed \n",
      "for\n",
      "             <b class=\"bold\">\n",
      "              the cloning of one of our plasmids\n",
      "             </b>\n",
      "             , in which we used different cloning strategies.  Each stage of the DBTL cycle is accompanied by an elaborated \n",
      " explanation, including results, thoughts, and decisions.\n",
      "            </p>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "       </div>\n",
      "       <div>\n",
      "       </div>\n",
      "      </div>\n",
      "     </div>\n",
      "     <div class=\"cropped-flipped\">\n",
      "      <img class=\"lower-curve\" src=\"https://static.igem.wiki/teams/4160/wiki/general/lower-background.svg\">\n",
      "       <img class=\"dashed-line-divider\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/dashedline-green-single.svg\">\n",
      "       </img>\n",
      "      </img>\n",
      "     </div>\n",
      "     <div class=\"lab_div_timeline\">\n",
      "      <div class=\"container\" style=\"margin-top: -30%;\">\n",
      "       <div class=\"timeline\">\n",
      "        <!-- Cycle 1 -->\n",
      "        <div class=\"row justify-content-center\" style=\"z-index:2;\">\n",
      "         <div class=\"col-6 col-md-4 col-lg-2 justify-content-center milestone_div order-3 order-lg-2\">\n",
      "          <image class=\"milestone-image\" src=\"https://static.igem.wiki/teams/4160/wiki/hp/golden-hp/reflectiepunt-blue.svg\"/>\n",
      "          <h2 class=\"numbered-milestone\">\n",
      "           1.\n",
      "          </h2>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 1 - Research -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               RESEARCH\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               In our project, we wanted to determine if antibody-induced GEMS receptor activation is \n",
      "possible. We hypothesized that by introducing an antigen on the EpoR receptor scaffold as an \n",
      "affinity domain, antibodies could activate the receptor. This activation would eventually lead\n",
      " to the expression of our therapeutically relevant protein, interleukin 10 (IL-10) (\n",
      "               <b>\n",
      "                Figure 1\n",
      "               </b>\n",
      "               ).\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab1\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 2 - Design-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               DESIGN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Obtaining the DNA that encodes for the longer aa linker length was relatively easy, as these plasmids\n",
      " were already present in the lab storage. One of these plasmids, already including our receptor backbone, \n",
      " coupled to a longer aa linker, was 50-CC (a receptor construct which uses a coiled coil (CC) as affinity \n",
      " domain). This plasmid stated to have a GEMS receptor construct that included an affinity domain that was \n",
      " not of interest to us, however, it was coupled to a 50 aa linker. As this linker length was of interest to\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab2\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 3 - Build -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               BUILD\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Building the new receptor construct with PR3 as an affinity domain with the 50 aa acid linker was \n",
      "expected to be relatively easy. After digestion by the restriction enzymes, an agarose gel was used to\n",
      " separate the digested products (\n",
      "               <b>\n",
      "                Figure 5\n",
      "               </b>\n",
      "               ). Plasmid length after digestion for 25(GS)_P3 is all around\n",
      " 7000 bp, and PR3 is 663 bp. The bands that are outlined were cut out and DNA was\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab3\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 4 - Test-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               TEST\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               To test if the cloning was successful, we transfected the ligated plasmid in TOP10 chemically competent\n",
      " cells. In addition, a negative control was added (only the digested plasmid, no insert) to check for false \n",
      " positives. Transformants were plated on LB-agar plates and grew overnight at 37 à¥°C.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab4\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 5 - LEARN -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               LEARN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Taking the sequencing results into account, we hypothesized that the plasmid we started with, did not include\n",
      " the sequence that we expected. Both the anticipated affinity domain and the 50 aa linker were not present in the plasmid.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab5\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Cycle 2 -->\n",
      "        <div class=\"row justify-content-center\" style=\"z-index:2;\">\n",
      "         <div class=\"col-6 col-md-4 col-lg-2 justify-content-center milestone_div order-3 order-lg-2\">\n",
      "          <image class=\"milestone-image\" src=\"https://static.igem.wiki/teams/4160/wiki/hp/golden-hp/reflectiepunt-blue.svg\"/>\n",
      "          <h2 class=\"numbered-milestone\">\n",
      "           2.\n",
      "          </h2>\n",
      "         </div>\n",
      "        </div>\n",
      "        <div>\n",
      "        </div>\n",
      "        <!-- Timeline block 5a - Research -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               RESEARCH\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               As the sequencing results were not as expected, we used commercial Sanger sequencing to \n",
      "                      determine the original DNA sequence of 25(GS)_P3. From this sequencing of the original plasmid, we \n",
      "                      concluded that the linker used in this plasmid consisted of 31 amino acids instead of the\n",
      "                       50 aa expected. These 31 amino acids are mostly GSS repeats with some other amino acids \n",
      "                       in between.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab5a\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 6 - Design-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               DESIGN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Therefore, we had to design a new cloning strategy to avoid the problems which are caused\n",
      " by using BamHI to digest the plasmid. Figure X shows a graphical overview of our new cloning\n",
      " strategy. It makes use of three different restriction sites, EcoRI, BamHI, and NotI, which \n",
      " cleaved the plasmid and PR3 insert in several ways. First of all, the original plasmid was \n",
      " cut with NotI and BamHI, which gives us the plasmid backbone including the ampicillin and \n",
      " kanamycin resistance. Next, the original plasmid was cut with EcoRI and NotI, resulting in\n",
      " the linker coupled to the receptor (without affinity domain). Finally, PR3 is digested using\n",
      " EcoRI and BamHI (\n",
      "               <b>\n",
      "                Figure 9\n",
      "               </b>\n",
      "               ). If these three parts were ligated correctly, the 31_PR3\n",
      " plasmid should be obtained.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab6\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 7 - Build -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               BUILD\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               To acquire this 31_PR3 receptor construct, digestions were performed on the original plasmid as\n",
      " described in the cloning strategy above. Digested products were analyzed and separated using an \n",
      " agarose gel (\n",
      "               <b>\n",
      "                Figure 10\n",
      "               </b>\n",
      "               ). The outlined DNA bands were cut out and purified using a gel \n",
      " purification kit from Qiagen. PR3 was digested with  BamHI and EcoRI. Clear, expected bands \n",
      " were seen at 5600 bp (25(GS)_P3 BB), around 1687 bp (EpoR + Linker), and 663 bp (PR3). The outlined\n",
      " DNA bands were cut out and purified.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab7\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 8 - Test-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               TEST\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               To test if the cloning was successful, we transfected the ligated plasmid in TOP10\n",
      " chemically competent cells. Moreover, a negative control was added (only the digested \n",
      " plasmid, no insert) to check for false positives. To test if the transformations work,\n",
      " a plasmid that we are confident works was inserted as a positive control. Transformants\n",
      " were plated on LB-agar plates and grew overnight at 37 à¥°C\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab8\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 9 - LEARN -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               LEARN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               We repeated all of the digestion and ligation steps, using fresh stocks for every step to \n",
      "ensure that the unsuccessful closing was not affected by any unexpected component in our old \n",
      "stocks. Unfortunately, this did not lead to successful cloning and transformation, as again no\n",
      " cultures were observed.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab9\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Cycle 3 -->\n",
      "        <div class=\"row justify-content-center\" style=\"z-index:2;\">\n",
      "         <div class=\"col-6 col-md-4 col-lg-2 justify-content-center milestone_div order-3 order-lg-2\">\n",
      "          <image class=\"milestone-image\" src=\"https://static.igem.wiki/teams/4160/wiki/hp/golden-hp/reflectiepunt-blue.svg\"/>\n",
      "          <h2 class=\"numbered-milestone\">\n",
      "           3.\n",
      "          </h2>\n",
      "         </div>\n",
      "        </div>\n",
      "        <div>\n",
      "        </div>\n",
      "        <!-- Timeline block 10 - Design-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               DESIGN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Unfortunately, the attempted cloning strategy which was discussed in the previous iteration\n",
      " of the DBTL cycle did not succeed. We intended to test an alternative cloning method utilizing\n",
      " a mechanism other than restriction and ligation because this cloning method had failed twice. \n",
      " One cloning method, which is based on a different mechanism, was Gibson assembly. Four new primers \n",
      " were designed (iGEM 016-019), which enabled overlap extension PCR used on the PR3 gBlocksTM gene \n",
      " fragment, and plasmid linearization using PCR of 25(GS)_P3. The latter simultaneously removed the original \n",
      " affinity domain from the 25(GS)_P3 plasmid. The full Gibson assembly strategy is depicted graphically in\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab10\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 11 - Build -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               BUILD\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               To obtain the 31_PR3 plasmid using Gibson assembly, two PCR reactions were performed.\n",
      " We performed overlap extension PCR to attach complementary Gibson assembly sites that \n",
      " facilitate the insertion of PR3 into the plasmid. Plasmid linearization was performed to\n",
      " remove the original affinity domain and simultaneously prime the plasmid for Gibson assembly.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab11\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 12 - Test-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               TEST\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Because we were not able to obtain the full 31_PR3 plasmid using this cloning technique,\n",
      " the â€œTestâ€ step was omitted in this iteration of the DBTL cycle.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab12\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 13 - LEARN -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               LEARN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Therefore, we discontinued using Gibson assembly to obtain the 31_PR3 plasmid.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab13\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Cycle 4 -->\n",
      "        <div class=\"row justify-content-center\" style=\"z-index:2;\">\n",
      "         <div class=\"col-6 col-md-4 col-lg-2 justify-content-center milestone_div order-3 order-lg-2\">\n",
      "          <image class=\"milestone-image\" src=\"https://static.igem.wiki/teams/4160/wiki/hp/golden-hp/reflectiepunt-blue.svg\"/>\n",
      "          <h2 class=\"numbered-milestone\">\n",
      "           4.\n",
      "          </h2>\n",
      "         </div>\n",
      "        </div>\n",
      "        <div>\n",
      "        </div>\n",
      "        <!-- Timeline block 14 - Design-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               DESIGN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Gibson assembly showed to be unsuccessful, depicted by the formation of unexpected PCR products.\n",
      " We had to design a new cloning strategy. Analysis of the 25(GS)_P3 sequence revealed a restriction site \n",
      " encoding for KlfI. As this restriction site is located inside the Igk secretion signal in front of \n",
      " the affinity domain, which is necessary for the transport of the receptor onto the membrane, a new\n",
      " cloning strategy is designed. This new cloning design uses overlap extension PCR to introduce a KlfI\n",
      " restriction site next to the PR3. Moreover, the primer is designed in such a way, that the Igk secretion\n",
      " signal does not get altered\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab14\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 15 - Build -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               BUILD\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Subsequent to receiving the new iGEM 020 primer,\n",
      " an overlap extension PCR reaction on the PR3 gBlock was performed. This should introduce a KlfI site in front of PR3.\n",
      " Hereafter, digestion of the plasmid and gBlocksTM gene fragment was done using the restriction enzymes KlfI and EcoRI.\n",
      " As many of our cloning techniques using PCR were not successful, we wanted to verify if this PCR reaction was successful.\n",
      " An analysis of the samples after PCR was performed using an agarose gel\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab15\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 16 - Test-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               TEST\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Despite the uncertaintly whether the ligation of the first overlap extension PCR was successful,\n",
      "we still wanted to verify if a full plasmid was created. This was done by transforming the ligation \n",
      "product in TOP10 chemically competent cells. Unfortunately, after an overnight incubation step of the \n",
      "transformants, no cultures were seen on the LB-Agar plate.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab16\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 17 - LEARN -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               LEARN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Both the gels used to analyze the digested products after overlap extension PCR and the\n",
      " transformation results suggest that the ligation was not successful. As we used overlap \n",
      " extension PCR to introduce a new restriction site into the gBlocksTM gene fragment, it was\n",
      " expected to see a very clear band around 750 bp as PCR duplicates this DNA exponentially. \n",
      " However, the first gel shows a smear in the lanes of PR3, indicating the presence of many \n",
      " differently-sized PCR products.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab17\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Cycle 5 -->\n",
      "        <div class=\"row justify-content-center\" style=\"z-index:2;\">\n",
      "         <div class=\"col-6 col-md-4 col-lg-2 justify-content-center milestone_div order-3 order-lg-2\">\n",
      "          <image class=\"milestone-image\" src=\"https://static.igem.wiki/teams/4160/wiki/hp/golden-hp/reflectiepunt-blue.svg\"/>\n",
      "          <h2 class=\"numbered-milestone\">\n",
      "           5.\n",
      "          </h2>\n",
      "         </div>\n",
      "        </div>\n",
      "        <div>\n",
      "        </div>\n",
      "        <!-- Timeline block 18 - Design-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               DESIGN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Because time was running out to obtain the 31 amino acid receptor construct and wanted to introduce\n",
      " an HA tag next to the original affinity domain, we ordered\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab18\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 19 - Build -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               BUILD\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               After receiving and completing the preparation of the new gBlock, the aforementioned cloning \n",
      "strategy is performed. Digestion of the 25(GS)_P3 plasmid and PR3 gBlock was performed using the \n",
      "restriction enzymes KlfI and EcoRI. To validate that the restriction happened correctly and \n",
      "subsequently separate the correct products, digestion products were put on an agarose gel\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab19\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 20 - Test-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               TEST\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               To validate if the cloning was successful, ligated products were transformed into TOP10 \n",
      "chemically competent cells. Unfortunately, the transformation was again unsuccessful as no\n",
      " cultures were present on the LB-Agar plate\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"lab20\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 21 - LEARN -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               LEARN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               We hypothesize that the unsuccessful cloning can be due to an undigested plasmid. \n",
      "Analysis of the gel seen in\n",
      "               <b>\n",
      "                Figure 20\n",
      "               </b>\n",
      "               (see Build) shows only undigested plasmid. A band at\n",
      "around 270 bp was expected, which is not seen in the gel. This indicates that the plasmid\n",
      "was either not digested or by only one of the two restriction enzymes.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"lab21\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div id=\"overlay\">\n",
      "      </div>\n",
      "      <!-- Cycle 1 -->\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup1\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose1\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/research-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Research\n",
      "        </h3>\n",
      "        <p>\n",
      "         In our project, we wanted to determine if antibody-induced GEMS receptor activation is \n",
      "possible. We hypothesized that by introducing an antigen on the EpoR receptor scaffold as an \n",
      "affinity domain, antibodies could activate the receptor. This activation would eventually lead\n",
      " to the expression of the therapeutically relevant protein, interleukin 10 (IL-10) (\n",
      "         <b class=\"bold\">\n",
      "          Figure 1\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/1.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 1 | Graphical representation of antibody-induced GEMS \n",
      "receptor activation, resulting in the expression of a therapeutically relevant protein.\n",
      "         </b>\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <p>\n",
      "         As described on the\n",
      "         <a class=\"link\" href=\" https://2022.igem.wiki/tu-eindhoven/results\" target=\"_blank\">\n",
      "          results\n",
      "         </a>\n",
      "         and\n",
      "         <a class=\"link\" href=\" https://2022.igem.wiki/tu-eindhoven/description\" target=\"_blank\">\n",
      "          project description\n",
      "         </a>\n",
      "         wiki pages, we intended to test differently sized amino acids linkers, which were introduced between the \n",
      "antigen (affinity domain) and EpoR receptor scaffold. As it is known that antibodies bind bivalently to \n",
      "antigens which are separated between 30 and 170 Ã… (with an optimal separation of 160 Ã…), we sought to create\n",
      " amino acid linker lengths which reside in this range.\n",
      "         <a class=\"references\" href=\"#references\" onclick=\"openReferences()\">\n",
      "          <sup>\n",
      "           1\n",
      "          </sup>\n",
      "         </a>\n",
      "         Besides\n",
      " the use of a 0 and 8 amino acid linker, with\n",
      "         <a class=\"link\" href=\"https://chemeslab.myqnapcloud.com:3939/CeffApp/\" target=\"_blank\">\n",
      "          lengths\n",
      "         </a>\n",
      "         of 0 Ã… and 14 Ã… respectively,\n",
      " the introduction of a longer amino acid (aa) linker would result in a more flexible linker. Click on the word lengths to go\n",
      " the online calculator we used to determine these linker lengths! The calculation is based on the worm-like chain model and empiric observations. We hypothesized that\n",
      " this longer amino acid linker represented the optimal antigen separation for antibody binding more closely. As all three\n",
      " linker variants were already present in plasmids found in the lab storage of the\n",
      "         <a class=\"link\" href=\"https://2022.igem.wiki/tu-eindhoven/attributions\" target=\"_blank\">\n",
      "          Synthetic Biology research group\n",
      "         </a>\n",
      "         at our university,\n",
      " the design of new linkers was not necessary. This page will only discuss the cloning strategies for obtaining 31_PR3, therefore,\n",
      " the information on the cloning of the 0 and 8 amino acid linker can be found in our\n",
      "         <a class=\"link\" href=\"https://2022.igem.wiki/tu-eindhoven/notebook\" target=\"_blank\">\n",
      "          Notebook\n",
      "         </a>\n",
      "         .\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup2\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose2\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/design-design-process.svg\"/>\n",
      "        <h3>\n",
      "         Design\n",
      "        </h3>\n",
      "        <p>\n",
      "         Obtaining the DNA that encodes for the longer 31 aa linker length was relatively easy, as these plasmids\n",
      " were already present in the lab storage. One of these plasmids, already including our receptor backbone, \n",
      " coupled to a longer aa linker, was 50-CC (a receptor construct which uses a coiled coil (CC) as affinity \n",
      " domain). This plasmid stated to have a GEMS receptor construct that included an affinity domain that was \n",
      " not of interest to us, however, it was coupled to a 50 aa linker. As this linker length was of interest to\n",
      " us, we sought to design a cloning strategy applicable to this plasmid.\n",
      "         <b class=\"bold\">\n",
      "          On this wiki page we will refer \n",
      " to this  plasmid as 25(GS)_P3\n",
      "         </b>\n",
      "         . The linker should cosists of 50 amino acids, which was a repeat of 25 times\n",
      " Glycine and Serine (GS). P3 refers to the type of coiled coil which is used as affinity domain.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         Although we had access to the plasmid encoding for the 50 amino acid linker, the exact DNA and amino acid\n",
      " sequence were unknown. We hypothesized that the amino acid sequence of this plasmid was consistent with the\n",
      " plasmid encoding for the 8 amino acid linker. We therefore simulated the 50 amino acid linker by multiplying\n",
      " the GS sequence in the 8 amino acid linker (\n",
      "         <b>\n",
      "          Figure 2\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/2.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 2 | Amino acid sequences of 8 aa linker and 50 aa linker.\n",
      "         </b>\n",
      "         Underlined amino acids encode for the EcoRI restriction site.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <p>\n",
      "         As stated in the project description, we intended to use these longer linkers between the EpoR part and the proteinase 3 (PR3) as an affinity domain (more explanation in the project description). \n",
      "The design of the PR3 affinity domain was made using a DNA sequence obtained from Uniprot (P24158). The N-terminal of PR3 was adapted to achieve correct folding. Finally,\n",
      " we introduced the restriction sites BamHI and EcoRI together with some extra bases, coding for amino acids, for stability next to the\n",
      " PR3 sequence (\n",
      "         <b class=\"bold\">\n",
      "          Figure 3\n",
      "         </b>\n",
      "         ). This sequence was ordered via IDT as a gBlocksTM gene fragment.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/3.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 3 | Amino acid sequence of the PR3 affinity domain.\n",
      "         </b>\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <p>\n",
      "         After constructing sequences for both the linker and the PR3 affinity domain, we were able to design a cloning\n",
      " strategy to acquire our receptor scaffold 25_PR3. By digesting both the plasmid and the gBlocksTM gene fragment \n",
      " with BamHI and EcoRI, followed by a ligation reaction using T4 ligase (NEB), the 25_PR3 receptor construct should\n",
      " be acquired (\n",
      "         <b class=\"bold\">\n",
      "          Figure 4\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/4.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 4 | Cloning strategy for obtaining 25_PR3.\n",
      "         </b>\n",
      "         By doing a 2-part \n",
      "ligation of the receptor + plasmid backbone and insert, 25_PR3 should be achieved.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup3\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose3\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/build-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Build\n",
      "        </h3>\n",
      "        <p>\n",
      "         Building the new receptor construct with PR3 as an affinity domain with the 50 aa linker was \n",
      "expected to be relatively easy. After digestion by the restriction enzymes, an agarose gel was used to\n",
      " separate the digested products (\n",
      "         <b class=\"bold\">\n",
      "          Figure 5\n",
      "         </b>\n",
      "         ). Bands are seen around 7000 kb, which depict the plasmid 25(GS)_P3 after digestion. The bands around 663 bp are the digested PR3. The bands that are outlined were cut out and DNA was purified out of the gel.\n",
      " The digested DNA fragments were stored over the weekend, after which ligation using T4 ligase was performed.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/5.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 5 | Agarose gel used for DNA separation of digested 25(GS)_P3 plasmid\n",
      " and PR3.\n",
      "         </b>\n",
      "         For digestion of the plasmid and insert BamHI (NEB) and EcoRI (NEB) were used. The outlined DNA\n",
      " bands were cut out and purified. Agarose gel (1%), ran for 1 hour at 100 V in 1x TAE buffer, stained with SYBR\n",
      " Safe DNA gel stain. Gel analysis was performed using a 470-nm blue light illuminator.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup4\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose4\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/test-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Test\n",
      "        </h3>\n",
      "        <p>\n",
      "         To test if the cloning was successful, we transfected the ligated plasmid in TOP10 chemically competent\n",
      " cells. In addition, a negative control was added (only the digested plasmid, no insert) to check for false \n",
      " positives. Transformants were plated on LB-agar plates and grew overnight at 37 à¥°C. The next day, colonies were\n",
      " seen to grow on the plate with the transformants (\n",
      "         <b class=\"bold\">\n",
      "          Figure 6\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         To verify that the cloning was successful, the plasmid was sequence verified using commercial Sanger sequencing \n",
      "(BaseClear). The most important sequencing results are shown in\n",
      "         <b class=\"bold\">\n",
      "          Figure 7\n",
      "         </b>\n",
      "         . The top line, which is partly not \n",
      "filled in with red, represents the missing DNA sequence. This missing DNA sequence would have encoded for the original affinity domain (purple), EcoRI restriction site (black), and the 50 aa linker (gray).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/6.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 6 | LB-Agar plates.\n",
      "         </b>\n",
      "         Top plate, TOP10 cells transformed with\n",
      " digested plasmid. The bottom plate, TOP10 cells transformed with the ligated 25_PR3 plasmid.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/7.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 7 | Sequencing data.\n",
      "         </b>\n",
      "         Sequencing data received after sending \n",
      "plasmid 25(GS)_P3 in for commercial Sanger sequencing (BaseClear). Sequencing primers iGEM 003 and iGEM 004 were \n",
      "used (\n",
      "         <a class=\"link\" href=\" https://2022.igem.wiki/tu-eindhoven/primers_gblocks\" target=\"_blank\">\n",
      "          Primer list\n",
      "         </a>\n",
      "         ). The top \n",
      "bar represents received sequencing data, with the non-filled in part representing a missing DNA sequence \n",
      "(bp ~3825 to 4075), when compared to the designed sequence.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup5\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose5\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/learn-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Learn\n",
      "        </h3>\n",
      "        <p>\n",
      "         As seen in\n",
      "         <b class=\"bold\">\n",
      "          Figure 6\n",
      "         </b>\n",
      "         , both the transformants with ligated plasmid and control transformants plate \n",
      "have grown. Cultures seen on the control plate were not expected to grow, as only the (non-ligated) digested \n",
      "plasmid was transformed in these cells. A possible reasoning for these results could be that one or both of the \n",
      "restriction enzymes were not successful in cleaving the plasmid, leaving the original plasmid unchanged.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         Taking the sequencing results into account, we hypothesized that the plasmid we started with, did not include\n",
      " the sequence that we expected. Both the anticipated affinity domain and the 50 aa linker were not present in the plasmid.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <!-- Cycle 2-->\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup5a\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose5a\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/research-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Research\n",
      "        </h3>\n",
      "        <p>\n",
      "         As the sequencing results were not as expected, we used commercial Sanger sequencing to \n",
      "determine the original DNA sequence of 25(GS)_P3. From this sequencing of the original plasmid, we \n",
      "concluded that the linker used in this plasmid consisted of 31 amino acids instead of the\n",
      " 50 aa expected. These 31 amino acids are mostly GSS repeats with some other amino acids \n",
      " in between (\n",
      "         <b class=\"bold\">\n",
      "          Figure 8\n",
      "         </b>\n",
      "         ). After analysis of the bases which encoded for this amino\n",
      " acids sequence, multiple BamHI restriction sites were found inside this linker, which is \n",
      " depicted by the underlined amino acids.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/8.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 8 | Amino acid sequence comparison.\n",
      "         </b>\n",
      "         Underlined \n",
      "amino acids show restriction sites for BamHI (GS) and EcoRI (EF).\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <p>\n",
      "         Our initial cloning method failed because the BamHI restriction sites appeared more than once \n",
      "in the sequence. The plasmid may be cut at several sites, which after ligation, can produce a \n",
      "variety of ligation products.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup6\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose6\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/design-design-process.svg\"/>\n",
      "        <h3>\n",
      "         Improved design\n",
      "        </h3>\n",
      "        <p>\n",
      "         Therefore, we had to design a new cloning strategy to avoid the problems which are caused\n",
      " by using BamHI to digest the plasmid. Figure 9 shows a graphical overview of our new cloning\n",
      " strategy. It makes use of three different restriction sites, EcoRI, BamHI, and NotI, which \n",
      " cleaved the plasmid and PR3 insert in several ways. First of all, the original plasmid was \n",
      " cut with NotI and BamHI, which gives us the plasmid backbone including the ampicillin and \n",
      " kanamycin resistance. Next, the original plasmid was cut with EcoRI and NotI, resulting in\n",
      " the linker coupled to the receptor (without affinity domain). Finally, PR3 is digested using\n",
      " EcoRI and BamHI (\n",
      "         <b class=\"bold\">\n",
      "          Figure 9\n",
      "         </b>\n",
      "         ). If these three parts were ligated correctly, the 31_PR3\n",
      " plasmid should be obtained.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/9.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 9 | New cloning strategy.\n",
      "         </b>\n",
      "         By performing a 3-part \n",
      "ligation of the receptor backbone, plasmid backbone, and insert, 31_PR3 should be achieved.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup7\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose7\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/build-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Build\n",
      "        </h3>\n",
      "        <p>\n",
      "         To acquire this 31_PR3 receptor construct, digestions were performed on the original plasmid as\n",
      " described in the cloning strategy above. Digested products were analyzed and separated using an \n",
      " agarose gel (\n",
      "         <b class=\"bold\">\n",
      "          Figure 10\n",
      "         </b>\n",
      "         ). The outlined DNA bands were cut out and purified using a gel \n",
      " purification kit from Qiagen. PR3 was digested with  BamHI-HF and EcoRI-HF. Clear, expected bands \n",
      " were seen at 5600 bp (25(GS)_P3 BB), around 1687 bp (EpoR + Linker), and 663 bp (PR3). The outlined\n",
      " DNA bands were cut out and purified.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/10.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 10 | Agarose gels for DNA separation of digested 25(GS)_P3, and PR3.\n",
      "         </b>\n",
      "         For\n",
      " the digestion of the plasmid backbone (25(GS)_P3 BB), NotI (NEB) and BamHI-HF (NEB) were used. For digestion of receptor\n",
      " backbone (EpoR + linker), NotI (NEB) and EcoRI-HF (NEB) were used. For digestion of PR3 BamHI (NEB) and EcoRI (NEB)\n",
      " were used. Agarose gel (1%), ran for 1 hour at 100 V in 1x TAE buffer, stained with SYBR Safe DNA gel stain. Gel \n",
      " analysis was performed using a 470-nm blue light illuminator.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <p>\n",
      "         The 3-part ligation was performed using T4 ligase enzyme in a 1:1:3, receptor backbone:plasmid backbone:insert.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup8\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose8\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/test-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Test\n",
      "        </h3>\n",
      "        <p>\n",
      "         To test if the cloning was successful, we transfected the ligated plasmid in TOP10\n",
      " chemically competent cells. Moreover, a negative control was added (only the digested \n",
      " plasmid, no insert) to check for false positives. To test if the transformations work,\n",
      " a plasmid that we are confident works was inserted as a positive control. Transformants\n",
      " were plated on LB-agar plates and grew overnight at 37 à¥°C (\n",
      "         <b class=\"bold\">\n",
      "          Figure 11\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"w-100\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/11.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 11 | LB-Agar plates.\n",
      "         </b>\n",
      "         TOP10 cells transformed with (f.l.t.r.)\n",
      " negative control ligation, ligation product, and positive control plasmid. Cultures only grew on the positive\n",
      " control plate.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup9\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose9\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/learn-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Learn\n",
      "        </h3>\n",
      "        <p>\n",
      "         As seen in\n",
      "         <b class=\"bold\">\n",
      "          Figure 11\n",
      "         </b>\n",
      "         , the plate containing the negative control and ligated transformants,\n",
      " no cultures could be detected the next day. However, the transformation was effective because cultures \n",
      " grew on the positive control plate. Therefore we hypothesized that either the restriction or ligation step \n",
      " was unsuccessful.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         We repeated all of the digestion and ligation steps, using fresh stocks for every step to \n",
      "ensure that the unsuccessful closing was not affected by any unexpected component in our old \n",
      "stocks. Unfortunately, this did not lead to successful cloning and transformation, as again no\n",
      " cultures were observed.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <!-- Cycle 3-->\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup10\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose10\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/design-design-process.svg\"/>\n",
      "        <h3>\n",
      "         Improved Design\n",
      "        </h3>\n",
      "        <p>\n",
      "         Unfortunately, the attempted cloning strategy which was discussed in the previous iteration\n",
      " of the DBTL cycle did not succeed. We intended to test an alternative cloning method utilizing\n",
      " a mechanism other than restriction and ligation because this cloning method had failed twice. \n",
      " One cloning method, which is based on a different mechanism, was Gibson assembly. Four new primers \n",
      " were designed (\n",
      "         <a class=\"link\" href=\" https://2022.igem.wiki/tu-eindhoven/primers_gblocks\" target=\"_blank\">\n",
      "          iGEM 016-019\n",
      "         </a>\n",
      "         )\n",
      " , which enabled overlap extension PCR used on the PR3 gBlocksTM gene \n",
      " fragment, and plasmid linearization using PCR of 25(GS)_P3. The latter simultaneously removed the original \n",
      " affinity domain from the 25(GS)_P3 plasmid. The full Gibson assembly strategy is depicted graphically in\n",
      "         <b class=\"bold\">\n",
      "          Figure 12\n",
      "         </b>\n",
      "         .\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/12.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 12 | Gibson cloning strategy.\n",
      "         </b>\n",
      "         The figure shows the steps\n",
      " taken to clone PR3 into the 25(GS)_P3 using Gibson assembly.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup11\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose11\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/build-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Build\n",
      "        </h3>\n",
      "        <p>\n",
      "         To obtain the 31_PR3 plasmid using Gibson assembly, two PCR reactions were performed.\n",
      " We performed overlap extension PCR to attach complementary Gibson assembly sites that \n",
      " facilitate the insertion of PR3 into the plasmid. Plasmid linearization was performed to\n",
      " remove the original affinity domain and simultaneously prime the plasmid for Gibson assembly.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         To verify if these PCR reactions were successful, samples of each PCR reaction were examined\n",
      " using an agarose gel (\n",
      "         <b class=\"bold\">\n",
      "          Figure 13\n",
      "         </b>\n",
      "         ). A DNA band are expected at ~6800 bp for the linearized \n",
      " plasmid and ~700 bp for PR3. No band is seen at the expected height for the linearized plasmid, \n",
      " which indicates that our PCR reaction was not successful.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/13.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 13| Agarose gel analysis on samples after PCR plasmid \n",
      "linearization and primer extension.\n",
      "         </b>\n",
      "         Samples after plasmid linearization (25(GS)_P3 (lin.)) and overlap \n",
      "extension PCR (PR3 (p.e.)). Agarose gel (1%), ran for 45 minutes at 120 V in 1x TAE buffer, stained with\n",
      " SYBR Safe DNA gel stain. Gel analysis was performed using a 470-nm blue light illuminator.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <p>\n",
      "         We repeated both PCR reactions, and again analyzed the samples on an agarose gel (\n",
      "         <b>\n",
      "          Figure 14\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/14.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 14 | Agarose gel analysis on samples after PCR plasmid \n",
      "linearization and primer extension.\n",
      "         </b>\n",
      "         Samples after plasmid linearization (25(GS)_P3 (lin.)) and overlap\n",
      " extension PCR (PR3 (p.e.)). Agarose gel (1%), ran for 45 minutes at 120 V in 1x TAE buffer, stained \n",
      " with SYBR Safe DNA gel stain. Gel analysis was performed using a 470-nm blue light illuminator.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup12\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose12\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/test-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Test\n",
      "        </h3>\n",
      "        <p>\n",
      "         Because we were not able to obtain the full 31_PR3 plasmid using this cloning technique,\n",
      " the â€œTestâ€ step was omitted in this iteration of the DBTL cycle.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup13\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose13\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/learn-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Learn\n",
      "        </h3>\n",
      "        <p>\n",
      "         The gels (\n",
      "         <b class=\"bold\">\n",
      "          Figures 13\n",
      "         </b>\n",
      "         and\n",
      "         <b class=\"bold\">\n",
      "          14\n",
      "         </b>\n",
      "         ) clearly indicate the formation of unintended PCR \n",
      "products during the plasmid linearization step. We hypothesized that the designed Gibson assembly\n",
      " primers have off-target binding sites. However, this assumption could not be confirmed by using\n",
      " in-silico methods to discover off-target binding sites.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         Therefore, we discontinued using Gibson assembly to obtain the 31_PR3 plasmid.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <!-- Cycle 4-->\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup14\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose14\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/design-design-process.svg\"/>\n",
      "        <h3>\n",
      "         Improved Design\n",
      "        </h3>\n",
      "        <p>\n",
      "         Gibson assembly showed to be unsuccessful, depicted by the formation of unexpected PCR products.\n",
      " We had to design a new cloning strategy. Analysis of the 25(GS)_P3 sequence revealed a restriction site \n",
      " encoding for Klf1. As this restriction site is located inside the Igk secretion signal in front of \n",
      " the affinity domain, which is necessary for the transport of the receptor onto the membrane, a new\n",
      " cloning strategy is designed. This new cloning design uses overlap extension PCR to introduce a Klf1\n",
      " restriction site next to the PR3. Moreover, the primer is designed in such a way, that the Igk secretion\n",
      " signal does not get altered (\n",
      "         <b class=\"bold\">\n",
      "          Figure 15\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/15.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 15 | New cloning strategy using overlap extension PCR.\n",
      "         </b>\n",
      "         By \n",
      "performing overlap extension PCR, a new Klf1 restriction site is introduced in the PR3 gBlock. Primer iGEM \n",
      "020 includes a part of the Igk secretion signal, to ensure that after ligation this tag stays intact.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup15\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose15\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/build-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Build\n",
      "        </h3>\n",
      "        <p>\n",
      "         Subsequent to receiving the new\n",
      "         <a class=\"link\" href=\" https://2022.igem.wiki/tu-eindhoven/primers_gblocks\" target=\"_blank\">\n",
      "          iGEM 020 primer\n",
      "         </a>\n",
      "         ,\n",
      " an overlap extension PCR reaction on the PR3 gBlock was performed. This should introduce a Klf1 site in front of PR3.\n",
      " Hereafter, digestion of the plasmid and gBlocksTM gene fragment was done using the restriction enzymes Klf1 and EcoRI-HF.\n",
      " As many of our cloning techniques using PCR were not successful, we wanted to verify if this PCR reaction was successful.\n",
      " An analysis of the samples after PCR was performed using an agarose gel (\n",
      "         <b class=\"bold\">\n",
      "          Figure 16\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/16.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 16 | Agarose gel analysis on samples after enzymatic digestion.\n",
      "         </b>\n",
      "         For \n",
      "the digestion of 25(GS)_P3 and PR3, Klf1 (ThermoFisher) and EcoRI-HF (NEB) were used. Agarose gel (1%), ran for 45 minutes\n",
      "at 120 V in 1x TAE buffer, stained with SYBR Safe DNA gel stain. Gel analysis was performed using a 470-nm blue light\n",
      " illuminator.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <p>\n",
      "         A clear band is seen at 6700 bp for linearized 25(GS)_P3. A very big smear is seen for the digested PR3 (expected 663 bp).\n",
      " As we were unsure if the band indicated a successful PCR reaction, extracted the DNA from this gel. An overnight ligation \n",
      " reaction was performed, resulting in the formation of 31_PR3. As it was uncertain whether the ligation worked, we repeated\n",
      " the overlap extension PCR followed by digestion by Klf1 and EcoRI-HF and analyzed the results again using an agarose gel\n",
      " (\n",
      "         <b class=\"bold\">\n",
      "          Figure 17\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/17.png \"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 17 | Agarose gel analysis on samples after enzymatic digestion.\n",
      "         </b>\n",
      "         For\n",
      " the digestion of 25(GS)_P3 and PR3, Klf1 (ThermoFisher) and EcoRI-HF (NEB) were used. Agarose gel (1%), ran for 45 minutes\n",
      " at 120 V in 1x TAE buffer, stained with SYBR Safe DNA gel stain. Gel analysis was performed using a 470-nm blue light\n",
      " illuminator.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup16\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose16\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/test-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Test\n",
      "        </h3>\n",
      "        <p>\n",
      "         Despite the uncertaintly whether the ligation of the first overlap extension PCR was successful,\n",
      "we still wanted to verify if a full plasmid was created. This was done by transforming the ligation \n",
      "product in TOP10 chemically competent cells. Unfortunately, after an overnight incubation step of the \n",
      "transformants, no cultures were seen on the LB-Agar plate.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup17\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose17\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/learn-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Learn\n",
      "        </h3>\n",
      "        <p>\n",
      "         Both the gels used to analyze the digested products after overlap extension PCR and the\n",
      " transformation results suggest that the ligation was not successful. As we used overlap \n",
      " extension PCR to introduce a new restriction site into the gBlocksTM gene fragment, it was\n",
      " expected to see a very clear band around 750 bp as PCR duplicates this DNA exponentially. \n",
      " However, the first gel shows a smear in the lanes of PR3, indicating the presence of many \n",
      " differently-sized PCR products.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         The second PCR reaction results, shown in\n",
      "         <b class=\"bold\">\n",
      "          Figure 17\n",
      "         </b>\n",
      "         (see Build), were not successful, as no bands can be \n",
      "seen in the wells of PR3.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         As we wanted to include an HA-tag in our receptor design as shown in\n",
      "         <b class=\"bold\">\n",
      "          Figure 18\n",
      "         </b>\n",
      "         (see Build), we\n",
      " needed to switch again from this cloning design to another.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/18.png \"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 18 | Graphic representation of receptor constructs 31_PR3 and 31_HA_PR3.\n",
      "         </b>\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <!-- Cycle 5-->\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup18\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose18\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/design-design-process.svg\"/>\n",
      "        <h3>\n",
      "         Improved Design\n",
      "        </h3>\n",
      "        <p>\n",
      "         Because time was running out to obtain the 31 amino acid receptor construct and wanted to introduce\n",
      " an HA tag next to the original affinity domain, we ordered\n",
      "         <a class=\"link\" href=\" https://2022.igem.wiki/tu-eindhoven/primers_gblocks\" target=\"_blank\">\n",
      "          a new PR3 gBlocksTM\n",
      "         </a>\n",
      "         gene fragment with an\n",
      " attached HA-tag. It also included the Klf1 tag including the part of the Igk secretion signal as described\n",
      " in our last cloning method. The cloning method we wanted to use is similar to the previous method, however,\n",
      " gBlocksTM gene fragments were ordered to obviate the overlap extension PCR step (\n",
      "         <b class=\"bold\">\n",
      "          Figure 19\n",
      "         </b>\n",
      "         ). \n",
      " This simplifies this cloning method to only a 2-part restriction and ligation cloning.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/19.png \"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 19 | Simplified cloning strategy to obtain 31_HA_PR3.\n",
      "         </b>\n",
      "         Both\n",
      " plasmid and gBlock are digested using Klf1 and EcoRI. Digested products are ligated using T4 ligase.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup19\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose19\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/build-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Build\n",
      "        </h3>\n",
      "        <p>\n",
      "         After receiving and completing the preparation of the new gBlock, the aforementioned cloning \n",
      "strategy is performed. Digestion of the 25(GS)_P3 plasmid and PR3 gBlock was performed using the \n",
      "restriction enzymes Klf1 and EcoRI. To validate that the restriction happened correctly and \n",
      "subsequently separate the correct products, digestion products were put on an agarose gel (\n",
      "         <b class=\"bold\">\n",
      "          Figure 20\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\" https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/20.png \"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 20 | Agarose gels for validation and DNA separation \n",
      "of digested plasmid and insert.\n",
      "         </b>\n",
      "         Digestion of HA_PR3 (718 bp) and 25(GS)_P3 (~6650 bp) by EcoRI-HF and Klf1. Agarose gel \n",
      "(1%), ran for 1 hour at 100 V in 1x TAE buffer, stained with SYBR Safe DNA gel stain. Gel analysis \n",
      "was performed using a 470-nm blue light illuminator.\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "        <p>\n",
      "         Outlined DNA was purified (Qiagen gel extraction kit), and ligated using T4 ligase (overnight at 4 à¥°C).\n",
      " This should provide us with the 31_HA_PR3 plasmid.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup20\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose20\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/test-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Test\n",
      "        </h3>\n",
      "        <p>\n",
      "         To validate if the cloning was successful, ligated products were transformed into TOP10 \n",
      "chemically competent cells. Unfortunately, the transformation was again unsuccessful as no\n",
      " cultures were present on the LB-Agar plate (\n",
      "         <b class=\"bold\">\n",
      "          Figure 21\n",
      "         </b>\n",
      "         ).\n",
      "        </p>\n",
      "        <br/>\n",
      "        <center>\n",
      "         <img class=\"image-tofull\" src=\"https://static.igem.wiki/teams/4160/wiki/lab/engi-suc/21.png\"/>\n",
      "        </center>\n",
      "        <figcaption style=\"font-size:13px\">\n",
      "         <b>\n",
      "          Figure 21 | LB-agar plate containing the TOP10 cells transformed \n",
      "with the ligated product.\n",
      "         </b>\n",
      "        </figcaption>\n",
      "        <br/>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"lab_popup21\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"lab_popupclose21\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/learn-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Learn\n",
      "        </h3>\n",
      "        <p>\n",
      "         We hypothesize that the unsuccessful cloning can be due to an undigested plasmid. \n",
      "Analysis of the gel seen in\n",
      "         <b class=\"bold\">\n",
      "          Figure 20\n",
      "         </b>\n",
      "         (see Build) shows only undigested plasmid. A band at\n",
      "around 270 bp was expected, which is not seen in the gel. This indicates that the plasmid\n",
      "was either not digested or by only one of the two restriction enzymes.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <!-- Disabling and enabling scroll background-->\n",
      "      <script>\n",
      "       function disable(){\n",
      "    document.querySelector('body').classList.add('disable-scroll');\n",
      "  }\n",
      "\n",
      "  function enable(){\n",
      "  document.querySelector('body').classList.remove('disable-scroll');\n",
      "}\n",
      "      </script>\n",
      "      <script>\n",
      "       var overlay = document.getElementById(\"overlay\");\n",
      "\n",
      "var lab_closePopup1 = document.getElementById(\"lab_popupclose1\");\n",
      "var lab_closePopup2 = document.getElementById(\"lab_popupclose2\");\n",
      "var lab_closePopup3 = document.getElementById(\"lab_popupclose3\");\n",
      "var lab_closePopup4 = document.getElementById(\"lab_popupclose4\");\n",
      "var lab_closePopup5 = document.getElementById(\"lab_popupclose5\");\n",
      "var lab_closePopup5a = document.getElementById(\"lab_popupclose5a\");\n",
      "var lab_closePopup6 = document.getElementById(\"lab_popupclose6\");\n",
      "var lab_closePopup7 = document.getElementById(\"lab_popupclose7\");\n",
      "var lab_closePopup8 = document.getElementById(\"lab_popupclose8\");\n",
      "var lab_closePopup9 = document.getElementById(\"lab_popupclose9\");\n",
      "var lab_closePopup10 = document.getElementById(\"lab_popupclose10\");\n",
      "var lab_closePopup11 = document.getElementById(\"lab_popupclose11\");\n",
      "var lab_closePopup12 = document.getElementById(\"lab_popupclose12\");\n",
      "var lab_closePopup13 = document.getElementById(\"lab_popupclose13\");\n",
      "var lab_closePopup14 = document.getElementById(\"lab_popupclose14\");\n",
      "var lab_closePopup15 = document.getElementById(\"lab_popupclose15\");\n",
      "var lab_closePopup16 = document.getElementById(\"lab_popupclose16\");\n",
      "var lab_closePopup17 = document.getElementById(\"lab_popupclose17\");\n",
      "var lab_closePopup18 = document.getElementById(\"lab_popupclose18\");\n",
      "var lab_closePopup19 = document.getElementById(\"lab_popupclose19\");\n",
      "var lab_closePopup20 = document.getElementById(\"lab_popupclose20\");\n",
      "var lab_closePopup21 = document.getElementById(\"lab_popupclose21\");\n",
      "\n",
      "var lab_popup1 = document.getElementById(\"lab_popup1\");\n",
      "var lab_popup2 = document.getElementById(\"lab_popup2\");\n",
      "var lab_popup3 = document.getElementById(\"lab_popup3\");\n",
      "var lab_popup4 = document.getElementById(\"lab_popup4\");\n",
      "var lab_popup5 = document.getElementById(\"lab_popup5\");\n",
      "var lab_popup5a = document.getElementById(\"lab_popup5a\");\n",
      "var lab_popup6 = document.getElementById(\"lab_popup6\");\n",
      "var lab_popup7 = document.getElementById(\"lab_popup7\");\n",
      "var lab_popup8 = document.getElementById(\"lab_popup8\");\n",
      "var lab_popup9 = document.getElementById(\"lab_popup9\");\n",
      "var lab_popup10 = document.getElementById(\"lab_popup10\");\n",
      "var lab_popup11 = document.getElementById(\"lab_popup11\");\n",
      "var lab_popup12 = document.getElementById(\"lab_popup12\");\n",
      "var lab_popup13 = document.getElementById(\"lab_popup13\");\n",
      "var lab_popup14 = document.getElementById(\"lab_popup14\");\n",
      "var lab_popup15 = document.getElementById(\"lab_popup15\");\n",
      "var lab_popup16 = document.getElementById(\"lab_popup16\");\n",
      "var lab_popup17 = document.getElementById(\"lab_popup17\");\n",
      "var lab_popup18 = document.getElementById(\"lab_popup18\");\n",
      "var lab_popup19 = document.getElementById(\"lab_popup19\");\n",
      "var lab_popup20 = document.getElementById(\"lab_popup20\");\n",
      "var lab_popup21 = document.getElementById(\"lab_popup21\");\n",
      "\n",
      "var lab1 = document.getElementById(\"lab1\");\n",
      "var lab2 = document.getElementById(\"lab2\");\n",
      "var lab3 = document.getElementById(\"lab3\");\n",
      "var lab4 = document.getElementById(\"lab4\");\n",
      "var lab5 = document.getElementById(\"lab5\");\n",
      "var lab5a = document.getElementById(\"lab5a\");\n",
      "var lab6 = document.getElementById(\"lab6\");\n",
      "var lab7 = document.getElementById(\"lab7\");\n",
      "var lab8 = document.getElementById(\"lab8\");\n",
      "var lab9 = document.getElementById(\"lab9\");\n",
      "var lab10 = document.getElementById(\"lab10\");\n",
      "var lab11 = document.getElementById(\"lab11\");\n",
      "var lab12 = document.getElementById(\"lab12\");\n",
      "var lab13 = document.getElementById(\"lab13\");\n",
      "var lab14 = document.getElementById(\"lab14\");\n",
      "var lab15 = document.getElementById(\"lab15\");\n",
      "var lab16 = document.getElementById(\"lab16\");\n",
      "var lab17 = document.getElementById(\"lab17\");\n",
      "var lab18 = document.getElementById(\"lab18\");\n",
      "var lab19 = document.getElementById(\"lab19\");\n",
      "var lab20 = document.getElementById(\"lab20\");\n",
      "var lab21 = document.getElementById(\"lab21\");\n",
      "\n",
      "// Show Overlay and Popup 1\n",
      "lab1.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup1.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event1 \n",
      "lab_closePopup1.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup1.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 2\n",
      "lab2.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup2.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 2\n",
      "lab_closePopup2.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup2.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 3\n",
      "lab3.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup3.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 3\n",
      "lab_closePopup3.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup3.style.display = 'none';\n",
      "  };\n",
      "  \n",
      "// Show Overlay and Popup 4\n",
      "lab4.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup4.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 4\n",
      "lab_closePopup4.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup4.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 5\n",
      "lab5.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup5.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 5\n",
      "lab_closePopup5.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup5.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 5a\n",
      "lab5a.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup5a.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 5a\n",
      "lab_closePopup5a.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup5a.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 6\n",
      "lab6.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup6.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 6\n",
      "lab_closePopup6.onclick = function() {\n",
      "  enable()  \n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup6.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 7\n",
      "lab7.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup7.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 7\n",
      "lab_closePopup7.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup7.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 8\n",
      "lab8.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup8.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 8\n",
      "lab_closePopup8.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup8.style.display = 'none';\n",
      "  };\n",
      "  \n",
      "// Show Overlay and Popup 9\n",
      "lab9.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup9.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 9\n",
      "lab_closePopup9.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup9.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 10\n",
      "lab10.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup10.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 10\n",
      "lab_closePopup10.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup10.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 11\n",
      "lab11.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup11.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event11 \n",
      "lab_closePopup11.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup11.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 12\n",
      "lab12.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup12.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 12\n",
      "lab_closePopup12.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup12.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 13\n",
      "lab13.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup13.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 13\n",
      "lab_closePopup13.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup13.style.display = 'none';\n",
      "  };\n",
      "  \n",
      "// Show Overlay and Popup 14\n",
      "lab14.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup14.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 14\n",
      "lab_closePopup14.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup14.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 15\n",
      "lab15.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup15.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 15\n",
      "lab_closePopup15.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup15.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 16\n",
      "lab16.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup16.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 16\n",
      "lab_closePopup16.onclick = function() {\n",
      "  enable()  \n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup16.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 17\n",
      "lab17.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup17.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 17\n",
      "lab_closePopup17.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup17.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 18\n",
      "lab18.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup18.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 18\n",
      "lab_closePopup18.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup18.style.display = 'none';\n",
      "  };\n",
      "  \n",
      "// Show Overlay and Popup 19\n",
      "lab19.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup19.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 19\n",
      "lab_closePopup19.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup19.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 20\n",
      "lab20.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup20.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 20\n",
      "lab_closePopup20.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup20.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 21\n",
      "lab21.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\tlab_popup21.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 20\n",
      "lab_closePopup21.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  lab_popup21.style.display = 'none';\n",
      "  };\n",
      "      </script>\n",
      "     </div>\n",
      "     <!--go to blue background-->\n",
      "     <div class=\"cropped\" style=\"margin-top:1em\">\n",
      "      <img class=\"dashed-line-divider flipped\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/dashedline-green-single.svg\">\n",
      "       <img class=\"backgroundcurves\" src=\"https://static.igem.wiki/teams/4160/wiki/general/upper-background.svg\">\n",
      "       </img>\n",
      "      </img>\n",
      "     </div>\n",
      "     <!-- Conclusion -->\n",
      "     <div class=\"layout\">\n",
      "      <!-- Start actual content of the page -->\n",
      "      <div class=\"blueblock outline\">\n",
      "       <div>\n",
      "       </div>\n",
      "       <div class=\"container container-custom\">\n",
      "        <div class=\"row mb-5 gx-0\" id=\"introduction\">\n",
      "         <div class=\"col-1 d-flex justify-content-center\">\n",
      "          <div class=\"bar-green\" style=\"height: 100%;\">\n",
      "          </div>\n",
      "         </div>\n",
      "         <div class=\"col-10\">\n",
      "          <div class=\"row gx-5\">\n",
      "           <div class=\"col-12 mb-3\">\n",
      "            <br/>\n",
      "            <h3>\n",
      "             <b class=\"bold\">\n",
      "              <center>\n",
      "               Conclusion\n",
      "              </center>\n",
      "             </b>\n",
      "            </h3>\n",
      "            <p>\n",
      "             <b class=\"bold\">\n",
      "              We have shown 5 iterations of the DBTL cycles\n",
      "             </b>\n",
      "             , which describe our process to obtain the\n",
      " 31_PR3 receptor scaffold. Unfortunately, even after all the cloning attempts which were \n",
      " based on different mechanisms, we were not able to successfully clone the affinity \n",
      " domain PR3 in the plasmid 25(GS)_P3.\n",
      "            </p>\n",
      "            <br/>\n",
      "            <p>\n",
      "             <b class=\"bold\">\n",
      "              However\n",
      "             </b>\n",
      "             , because obtaining a receptor scaffold including a longer amino acid linker \n",
      "is still of high importance for the future of !MPACT, we will end this DBTL cycle with \n",
      "suggestions that could result in the successful cloning of a longer linker length in a \n",
      "receptor construct.\n",
      "            </p>\n",
      "            <br/>\n",
      "            <p>\n",
      "             <b class=\"bold\">\n",
      "              We hypothesize that\n",
      "             </b>\n",
      "             using Gibson assembly cloning to obtain 31_PR3 has the most potential\n",
      " to succeed, as by using this technique you have complete control over what parts are inserted.\n",
      " By designing new Gibson primers specific for 25(GS)_P3 and ensuring they do not have off-target binding sites,\n",
      " we expect that the cloning could be done relatively easily.\n",
      "            </p>\n",
      "            <br/>\n",
      "            <p>\n",
      "             If problems still arise while using this method, other methods could include the use of a different \n",
      "starting plasmid (e.g. 4_P3). By using plasmid linearization with primers designed to remove the full \n",
      "affinity domain and original linker, a plasmid including the receptor backbone can be obtained. Next, a \n",
      "gBlocksTM gene fragment which encodes for the affinity domain of choice, coupled to the linker length of\n",
      " choice, can be ordered. Next, both fragments can be ligated togheter, resulting in the receptor construct with a longer linker length.\n",
      " If all fails, you could still contact IDT or Twist to synthesize a gene or even use their cloning service.\n",
      "            </p>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <div class=\"row mb-5 gx-0\" id=\"reference\">\n",
      "         <div class=\"col-1 d-flex justify-content-center\">\n",
      "         </div>\n",
      "         <div class=\"col-10\">\n",
      "          <div class=\"row gx-5\">\n",
      "           <div class=\"col-12 mb-3\">\n",
      "            <button class=\"accordionwhite\" id=\"reference\">\n",
      "             <h5>\n",
      "              References\n",
      "             </h5>\n",
      "             <div class=\"white-arrow-circle\">\n",
      "             </div>\n",
      "            </button>\n",
      "            <div class=\"panelwhite\">\n",
      "             <div style=\"padding:1.25em 0.5em;\">\n",
      "              <ol>\n",
      "               <li>\n",
      "                Shaw A, Hoffecker IT, Smyrlaki I, et al. Binding to nanopatterned antigens is dominated by the spatial tolerance of antibodies. Nat Nanotechnol. 2019;14(2):184-190. doi:10.1038/s41565-018-0336-3\n",
      "               </li>\n",
      "              </ol>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "       </div>\n",
      "       <div>\n",
      "       </div>\n",
      "      </div>\n",
      "     </div>\n",
      "    </div>\n",
      "    <div id=\"education_div\">\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:38%; left:25%; width:15%; -ms-transform: rotate(316deg); transform: rotate(316deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:44%; left:72%; width:30%; -ms-transform: rotate(160deg); transform: rotate(160deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:56%; left:2%; width:30%; -ms-transform: rotate(200deg); transform: rotate(200deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:64%; left:68%; width:10%; -ms-transform: rotate(30deg); transform: rotate(30deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:68%; left:70%; width:35%; -ms-transform: rotate(155deg); transform: rotate(155deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:74%; left:4%; width:22%; -ms-transform: rotate(-45deg); transform: rotate(-45deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:90%; left:88%; width:10%; -ms-transform: rotate(25deg); transform: rotate(25deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div>\n",
      "      <img class=\"background-c-image\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/background-c-white.png\" style=\"top:86%; left:65%; width:17%; -ms-transform: rotate(130deg); transform: rotate(130deg); z-index:-1;\"/>\n",
      "     </div>\n",
      "     <div class=\"layout\">\n",
      "      <!-- Start actual content of the page -->\n",
      "      <div class=\"blueblock outline\">\n",
      "       <div>\n",
      "       </div>\n",
      "       <div class=\"container container-custom\">\n",
      "        <div class=\"row mb-5 gx-0\" id=\"introduction\">\n",
      "         <div class=\"col-1 d-flex justify-content-center\">\n",
      "          <!--<div class=\"bar-green\" style=\"height: 100%;\"></div> -->\n",
      "         </div>\n",
      "         <div class=\"col-10\">\n",
      "          <div class=\"row gx-5\">\n",
      "           <div class=\"col-12 mb-3\">\n",
      "            <br/>\n",
      "            <br/>\n",
      "            <p>\n",
      "             In addition to our project, we focused on educating different target groups about synthetic biology. One of these target groups was elementary school children. For these children, we organized a DNA lesson, given at the elementary school in Haaren. More information about our experience and the lesson can be found on the\n",
      "             <a class=\"link\" href=\"https://2022.igem.wiki/tu-eindhoven/education\" target=\"_blank\">\n",
      "              education\n",
      "             </a>\n",
      "             page.\n",
      "            </p>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "       </div>\n",
      "       <div>\n",
      "       </div>\n",
      "      </div>\n",
      "     </div>\n",
      "     <div class=\"cropped-flipped\">\n",
      "      <img class=\"lower-curve\" src=\"https://static.igem.wiki/teams/4160/wiki/general/lower-background.svg\">\n",
      "       <img class=\"dashed-line-divider\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/dashedline-green-single.svg\">\n",
      "       </img>\n",
      "      </img>\n",
      "     </div>\n",
      "     <div class=\"lab_div_timeline\">\n",
      "      <div class=\"container\" style=\"margin-top: -30%;z-index:2\">\n",
      "       <div class=\"timeline\">\n",
      "        <!-- Cycle 1 -->\n",
      "        <div class=\"row justify-content-center\" style=\"z-index:2;\">\n",
      "         <div class=\"col-6 col-md-4 col-lg-2 justify-content-center milestone_div order-3 order-lg-2\">\n",
      "          <image class=\"milestone-image\" src=\"https://static.igem.wiki/teams/4160/wiki/hp/golden-hp/reflectiepunt-blue.svg\"/>\n",
      "          <h2 class=\"numbered-milestone\">\n",
      "           1.\n",
      "          </h2>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 1 - Research -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               RESEARCH\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Before we could start designing, more information was required. Such as, the age of the children and what do they already know about DNA. We asked the elementary school teacher for this information and learned the children would be about 10 to 11 years old. Furthermore, the school was in the middle of a project about DNA. Therefore, the students had already a bit of knowledge about DNA. This was implemented into the design.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"education1\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 2 - Design-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               DESIGN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               The teachers of these classes wanted a lesson about what DNA is, and what research you can do with DNA. This was an addition to a running project the school had; the teachers were looking for experts to tell them about synthetic biology. Therefore, the purpose of this lesson was to make them familiar with synthetic biology starting with the principles of DNA.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"education2\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 3 - Build -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               BUILD\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               The presentation was separated into two parts. The first part contained general information about DNA. Explained with the use of lots of simple icons. The second part explained the research which can be done with DNA. This was explained by using simple examples and again icons. The parts were separated by a small, interactive game, the DNA game, where the students could form a DNA sequence presented on the presentation with our pre-prepared nucleotide letter signs. We made signs with a cord with the letters A, T, G, and C (the nucleotides) which the students could use to form a DNA sequence of four nucleotide pairs.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"education3\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 4 - Design-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               TEST\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               Before performing the lesson on the 16th of May in Haaren, we tested it a couple of times. Doing so by presenting for each other and making sure we spoke in an easy and clear language. The real test was performed on the 16th of May for the students and teacher. We gave the presentation, explained everything clearly and understandable to the students, and the students played the game. \n",
      "We received a ton of questions from the children! Not only about the information we gave, but also questions that were built on the information we gave. So, we could conclude that they did understand everything we explained and wanted to learn even more about DNA.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"education4\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 5 - LEARN -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               LEARN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               After this lesson, we evaluated with our team, and we also received some feedback from the teacher.\n",
      "We learned that we handled the questions from the children well and were adapting to the level of the elementary school children. Furthermore, it was nice that there was a lot of time for questions. Although, when answering questions, we could say â€™I do not knowâ€™ if we do not know the answer to a question.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"education5\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Cycle 2 -->\n",
      "        <div class=\"row justify-content-center\" style=\"z-index:2;\">\n",
      "         <div class=\"col-6 col-md-4 col-lg-2 justify-content-center milestone_div order-3 order-lg-2\">\n",
      "          <image class=\"milestone-image\" src=\"https://static.igem.wiki/teams/4160/wiki/hp/golden-hp/reflectiepunt-blue.svg\"/>\n",
      "          <h2 class=\"numbered-milestone\">\n",
      "           2.\n",
      "          </h2>\n",
      "         </div>\n",
      "        </div>\n",
      "        <div>\n",
      "        </div>\n",
      "        <!-- Timeline block 6 - RESEARCH-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               RESEARCH\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               The next lesson would be given to the exact same age group as before, only at a different school. This time we would give our lesson in Uden to teach children about DNA. These children did not have any prior knowledge about DNA.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"education6\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 7 - DESIGN -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               DESIGN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               The design for the setup of this lesson is mostly the same, as there was no negative feedback about the presentation and information. Only the execution of the presentation and the game did change, taking in the points of improvement from the teacher as mentioned above.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"education7\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 8 - Build-->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               BUILD\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               The alterations in design were practiced by practicing the presentation again.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12 d-flex justify-content-end\">\n",
      "              <button class=\"readmore\" id=\"education8\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        <!-- Timeline block 9 - TEST -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               TEST\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               We performed the next lesson on the 20th of June in Uden. This time we made sure enough time was taken to explain everything clearly, since these children did not have prior knowledge. Besides, we implemented the feedback by explaining the game step by step and asking children one by one to the front of the classroom. Furthermore, when answering questions, we took a lot of time. This is the same as the previous time, since this was seen as very positive. Now, we also said that we did not know how to answer a question when we did not know.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"education9\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "        '\n",
      "        <!-- Timeline block 10 - LEARN -->\n",
      "        <div class=\"row no-gutters justify-content-end justify-content-md-between align-items-start timeline-nodes\">\n",
      "         <div class=\"col-10 col-md-5 order-3 order-md-1 timeline-content\">\n",
      "          <div class=\"timelineblock blue-timelineblock\">\n",
      "           <div class=\"container information\">\n",
      "            <div class=\"row gy-3\">\n",
      "             <div class=\"col-12\">\n",
      "              <h3>\n",
      "               LEARN\n",
      "              </h3>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <p class=\"cutoff-text\">\n",
      "               After this lesson we reflected again. We could happily conclude all the previous feedback points were implemented. From the teacher we received some new (positive) feedback as well. She mentioned the presentation had a clear structure and visualized our words. There was a lot of interaction with the children. Furthermore, she gave one tip for improvement. She mentioned it is a good idea to let the children do something themselves. Although she also mentioned that she enjoyed the interactive exercise we had thought about, there the children did something themselves as well.\n",
      "              </p>\n",
      "             </div>\n",
      "             <div class=\"col-12\">\n",
      "              <button class=\"readmore\" id=\"education10\">\n",
      "               Read more\n",
      "              </button>\n",
      "             </div>\n",
      "            </div>\n",
      "           </div>\n",
      "          </div>\n",
      "         </div>\n",
      "        </div>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div id=\"overlay\">\n",
      "      </div>\n",
      "      <!-- Cycle 1 -->\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup1\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose1\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/research-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Research\n",
      "        </h3>\n",
      "        <p>\n",
      "         Before we could start designing, more information was required. Such as, the age of the children and what do they already know about DNA. We asked the elementary school teacher for this information and learned the children would be about 10 to 11 years old. Furthermore, the school was in the middle of a project about DNA. Therefore, the students had already a bit of knowledge about DNA. This was implemented into the design.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup2\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose2\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/design-design-process.svg\"/>\n",
      "        <h3>\n",
      "         Design\n",
      "        </h3>\n",
      "        <p>\n",
      "         The teachers of these classes wanted a lesson about what DNA is, and what research you can do with DNA. This was an addition to a running project the school had; the teachers were looking for experts to tell them about synthetic biology. Therefore, the purpose of this lesson was to make them familiar with synthetic biology starting with the principles of DNA.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         After receiving the requirements of the teachers regarding the content of the lesson we started brainstorming. The proposed design is a presentation with a lot of easy and explaining images and an interactive game. With the game, the students would form their own short DNA sequence, by standing at the front of the class with a sign of a nucleotide around their neck and holding hands.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup3\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose3\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/build-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Build\n",
      "        </h3>\n",
      "        <p>\n",
      "         The presentation was separated into two parts. The first part contained general information about DNA. Explained with the use of lots of simple icons. The second part explained the research which can be done with DNA. This was explained by using simple examples and again icons. The parts were separated by a small, interactive game, the DNA game, where the students could form a DNA sequence presented on the presentation with our pre-prepared nucleotide letter signs. We made signs with a cord with the letters A, T, G, and C (the nucleotides) which the students could use to form a DNA sequence of four nucleotide pairs.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup4\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose4\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/test-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Test\n",
      "        </h3>\n",
      "        <p>\n",
      "         Before performing the lesson on the 16th of May in Haaren, we tested it a couple of times. Doing so by presenting for each other and making sure we spoke in an easy and clear language. The real test was performed on the 16th of May for the students and teacher. We gave the presentation, explained everything clearly and understandable to the students, and the students played the game.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         We received a ton of questions from the children! Not only about the information we gave, but also questions that were built on the information we gave. So, we could conclude that they did understand everything we explained and wanted to learn even more about DNA.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup5\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose5\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/learn-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Learn\n",
      "        </h3>\n",
      "        <p>\n",
      "         After this lesson, we evaluated with our team, and we also received some feedback from the teacher. \n",
      "We learned that we handled the questions from the children well and were adapting to the level of the elementary school children. Furthermore, it was nice that there was a lot of time for questions. Although, when answering questions, we could say â€™I do not knowâ€™ if we do not know the answer to a question.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         Also, the teacher mentioned the game was a nice active manner of teaching the principles of DNA. Although we should explain the game easier for the next group. Also, it might work better if we would choose one child at a time when bringing the children to the front of the class for the game. The teacher also mentioned that this DNA lesson was a good addition to the knowledge that the students already had. We had a powerful attitude in front of the class and were enthusiastic about the subject. Although, the next time we should talk slower.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <!-- Cycle 2-->\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup6\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose6\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/research-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Research\n",
      "        </h3>\n",
      "        <p>\n",
      "         The next lesson would be given to the exact same age group as before, only at a different school. This time we would give our lesson in Uden to teach children about DNA. These children did not have any prior knowledge about DNA.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup7\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose7\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/design-design-process.svg\"/>\n",
      "        <h3>\n",
      "         Design\n",
      "        </h3>\n",
      "        <p>\n",
      "         The design for the setup of this lesson is mostly the same, as there was no negative feedback about the presentation and information. Only the execution of the presentation and the game did change, taking in the points of improvement from the teacher as mentioned above.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         The plan is when the game starts to ask children one by one to the front of the class and take more time to explain the game step by step. Furthermore, when answering questions, we planned on taking enough time to answer them, and if we do not know the answer, we should say it.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup8\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose8\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/build-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Build\n",
      "        </h3>\n",
      "        <p>\n",
      "         The alterations in design were practiced by practicing the presentation again.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup9\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose9\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/test-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Test\n",
      "        </h3>\n",
      "        <p>\n",
      "         We performed the next lesson on the 20th of June in Uden. This time we made sure enough time was taken to explain everything clearly, since these children did not have prior knowledge. Besides, we implemented the feedback by explaining the game step by step and asking children one by one to the front of the classroom. Furthermore, when answering questions, we took a lot of time. This is the same as the previous time, since this was seen as very positive. Now, we also said that we did not know how to answer a question when we did not know.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         We gave the lesson with the same enthusiasm as the previous time. Also, this time the lesson was received positively, again the children asked tons of questions.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div class=\"project-popup blue-popup\" id=\"education_popup10\">\n",
      "       <div class=\"popupcontrols\">\n",
      "        <span class=\"popupclose\" id=\"education_popupclose10\">\n",
      "         X\n",
      "        </span>\n",
      "       </div>\n",
      "       <div class=\"project-popup-content\">\n",
      "        <img class=\"w-100\" src=\"https://static.igem.wiki/teams/4160/wiki/engineering-succes/learn-design-proces.svg\"/>\n",
      "        <h3>\n",
      "         Learn\n",
      "        </h3>\n",
      "        <p>\n",
      "         After this lesson, we evaluated with our team, and we also received some feedback from the teacher. \n",
      "We learned that we handled the questions from the children well and were adapting to the level of the elementary school children. Furthermore, it was nice that there was a lot of time for questions. Although, when answering questions, we could say â€™I do not knowâ€™ if we do not know the answer to a question.\n",
      "        </p>\n",
      "        <br/>\n",
      "        <p>\n",
      "         Also, the teacher mentioned the game was a nice active manner of teaching the principles of DNA. Although we should explain the game easier for the next group. Also, it might work better if we would choose one child at a time when bringing the children to the front of the class for the game. The teacher also mentioned that this DNA lesson was a good addition to the knowledge that the students already had. We had a powerful attitude in front of the class and were enthusiastic about the subject. Although, the next time we should talk slower.\n",
      "        </p>\n",
      "       </div>\n",
      "      </div>\n",
      "      <!-- Disabling and enabling scroll background-->\n",
      "      <script>\n",
      "       function disable(){\n",
      "    document.querySelector('body').classList.add('disable-scroll');\n",
      "  }\n",
      "\n",
      "  function enable(){\n",
      "  document.querySelector('body').classList.remove('disable-scroll');\n",
      "}\n",
      "      </script>\n",
      "      <script>\n",
      "       var overlay = document.getElementById(\"overlay\");\n",
      "\n",
      "var education_closePopup1 = document.getElementById(\"education_popupclose1\");\n",
      "var education_closePopup2 = document.getElementById(\"education_popupclose2\");\n",
      "var education_closePopup3 = document.getElementById(\"education_popupclose3\");\n",
      "var education_closePopup4 = document.getElementById(\"education_popupclose4\");\n",
      "var education_closePopup5 = document.getElementById(\"education_popupclose5\");\n",
      "var education_closePopup6 = document.getElementById(\"education_popupclose6\");\n",
      "var education_closePopup7 = document.getElementById(\"education_popupclose7\");\n",
      "var education_closePopup8 = document.getElementById(\"education_popupclose8\");\n",
      "var education_closePopup9 = document.getElementById(\"education_popupclose9\");\n",
      "var education_closePopup10 = document.getElementById(\"education_popupclose10\");\n",
      "\n",
      "var education_popup1 = document.getElementById(\"education_popup1\");\n",
      "var education_popup2 = document.getElementById(\"education_popup2\");\n",
      "var education_popup3 = document.getElementById(\"education_popup3\");\n",
      "var education_popup4 = document.getElementById(\"education_popup4\");\n",
      "var education_popup5 = document.getElementById(\"education_popup5\");\n",
      "var education_popup6 = document.getElementById(\"education_popup6\");\n",
      "var education_popup7 = document.getElementById(\"education_popup7\");\n",
      "var education_popup8 = document.getElementById(\"education_popup8\");\n",
      "var education_popup9 = document.getElementById(\"education_popup9\");\n",
      "var education_popup10 = document.getElementById(\"education_popup10\");\n",
      "\n",
      "var education1 = document.getElementById(\"education1\");\n",
      "var education2 = document.getElementById(\"education2\");\n",
      "var education3 = document.getElementById(\"education3\");\n",
      "var education4 = document.getElementById(\"education4\");\n",
      "var education5 = document.getElementById(\"education5\");\n",
      "var education6 = document.getElementById(\"education6\");\n",
      "var education7 = document.getElementById(\"education7\");\n",
      "var education8 = document.getElementById(\"education8\");\n",
      "var education9 = document.getElementById(\"education9\");\n",
      "var education10 = document.getElementById(\"education10\");\n",
      "\n",
      "// Show Overlay and Popup 1\n",
      "education1.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup1.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event1 \n",
      "education_closePopup1.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup1.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 2\n",
      "education2.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup2.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 2\n",
      "education_closePopup2.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup2.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 3\n",
      "education3.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup3.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 3\n",
      "education_closePopup3.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup3.style.display = 'none';\n",
      "  };\n",
      "  \n",
      "// Show Overlay and Popup 4\n",
      "education4.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup4.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 4\n",
      "education_closePopup4.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup4.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 5\n",
      "education5.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup5.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 5\n",
      "education_closePopup5.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup5.style.display = 'none';\n",
      "  };\n",
      "\n",
      "\n",
      "\n",
      "// Show Overlay and Popup 6\n",
      "education6.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup6.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 6\n",
      "education_closePopup6.onclick = function() {\n",
      "  enable()  \n",
      "  overlay.style.display = 'none';\n",
      "  education_popup6.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 7\n",
      "education7.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup7.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 7\n",
      "education_closePopup7.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup7.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 8\n",
      "education8.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup8.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 8\n",
      "education_closePopup8.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup8.style.display = 'none';\n",
      "  };\n",
      "  \n",
      "// Show Overlay and Popup 9\n",
      "education9.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup9.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 9\n",
      "education_closePopup9.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup9.style.display = 'none';\n",
      "  };\n",
      "\n",
      "// Show Overlay and Popup 10\n",
      "education10.onclick = function() {\n",
      "  disable()\n",
      "  overlay.style.display = 'block';\n",
      "\teducation_popup10.style.display = 'block';\n",
      "\t};\n",
      "// Close Popup Event 10\n",
      "education_closePopup10.onclick = function() {\n",
      "  enable()\n",
      "  overlay.style.display = 'none';\n",
      "  education_popup10.style.display = 'none';\n",
      "  };\n",
      "      </script>\n",
      "     </div>\n",
      "     <!--go to blue background-->\n",
      "     <div class=\"cropped\" style=\"margin-top:1em\">\n",
      "      <img class=\"dashed-line-divider flipped\" src=\"https://static.igem.wiki/teams/4160/wiki/general/miscellaneous/dashedline-green-single.svg\">\n",
      "       <img class=\"backgroundcurves\" src=\"https://static.igem.wiki/teams/4160/wiki/general/upper-background.svg\">\n",
      "       </img>\n",
      "      </img>\n",
      "     </div>\n",
      "     <!-- Conclusion -->\n",
      "     <div class=\"blueblock outline\">\n",
      "      <div>\n",
      "      </div>\n",
      "      <div class=\"container container-custom\">\n",
      "       <div class=\"row mb-5 gx-0\" id=\"aav\">\n",
      "        <div class=\"col-12\">\n",
      "        </div>\n",
      "       </div>\n",
      "      </div>\n",
      "      <div>\n",
      "      </div>\n",
      "     </div>\n",
      "    </div>\n",
      "    <script>\n",
      "     let lab = document.getElementById(\"lab_button\");\n",
      "let education = document.getElementById(\"education_button\");\n",
      "\n",
      "let lab_div = document.getElementById(\"lab_div\");\n",
      "let education_div = document.getElementById(\"education_div\");\n",
      "\n",
      "function openOtherDiv(clicked_div){\n",
      "  const selected_buttons = document.getElementsByClassName(\"subtitle-selected\")\n",
      "\n",
      "  if (selected_buttons.length > 0){\n",
      "    selected_buttons[0].classList.add(\"subtitle-unselected\")\n",
      "    selected_buttons[0].classList.remove(\"subtitle-selected\")\n",
      "  }\n",
      "\n",
      "  if(clicked_div === 'lab'){\n",
      "    lab_div.style.display = 'block'\n",
      "    education_div.style.display = 'none'\n",
      "    lab.classList.remove(\"subtitle-unselected\")\n",
      "    lab.classList.add(\"subtitle-selected\")\n",
      "\n",
      "  } else if (clicked_div === 'education'){\n",
      "    lab_div.style.display = 'none'\n",
      "    education_div.style.display = 'block'\n",
      "    \n",
      "    education.classList.remove(\"subtitle-unselected\")\n",
      "    education.classList.add(\"subtitle-selected\")\n",
      "  }\n",
      "};\n",
      "    </script>\n",
      "    <script>\n",
      "     var accwhite = document.getElementsByClassName(\"accordionwhite\");\n",
      "  var i;\n",
      "\n",
      "  for (i = 0; i < accwhite.length; i++) {\n",
      "      accwhite[i].addEventListener(\"click\", function () {\n",
      "          /* Toggle between adding and removing the \"active\" class,\n",
      "          to highlight the button that controls the panel */\n",
      "          this.classList.toggle(\"active\");\n",
      "\n",
      "          /* Toggle between hiding and showing the active panel */\n",
      "          var panel = this.nextElementSibling;\n",
      "          if (panel.style.maxHeight) {\n",
      "              panel.style.maxHeight = null;\n",
      "              \n",
      "\n",
      "          } else {\n",
      "              panel.style.maxHeight = '100%';\n",
      "          }\n",
      "         \n",
      "      });\n",
      "  }\n",
      "    </script>\n",
      "   </img>\n",
      "  </div>\n",
      "  <!-- Footer: MUST mention license AND have a link to team wiki's repository on gitlab.igem.org -->\n",
      "  <footer class=\"pt-5 pb-5 footer text-white footer-custom\">\n",
      "   <div class=\"container\">\n",
      "    <div class=\"row d-flex justify-content-center gy-3\">\n",
      "     <div class=\"col-lg-2 col-sm-4 col-6 d-flex flex-column align-items-center\">\n",
      "      <h4 class=\"mb-3\">\n",
      "       Postal adres\n",
      "      </h4>\n",
      "      <p>\n",
      "       iGEM TU Eindhoven\n",
      "      </p>\n",
      "      <p>\n",
      "       Postbox 513\n",
      "      </p>\n",
      "      <p>\n",
      "       5600MB Eindhoven\n",
      "      </p>\n",
      "      <p>\n",
      "       The Netherlands\n",
      "      </p>\n",
      "     </div>\n",
      "     <div class=\"col-lg-2 col-sm-4 col-6 d-flex flex-column align-items-center\">\n",
      "      <h4 class=\"mb-3\">\n",
      "       Office\n",
      "      </h4>\n",
      "      <p>\n",
      "       Eindhoven,\n",
      "      </p>\n",
      "      <p>\n",
      "       The Netherlands\n",
      "      </p>\n",
      "      <p>\n",
      "       TU/e campus\n",
      "      </p>\n",
      "      <p>\n",
      "       Traverse, 1.47 &amp; 1.48\n",
      "      </p>\n",
      "     </div>\n",
      "     <div class=\"col-lg-4 col-8 flex-column\">\n",
      "      <img alt=\"logo_igem\" src=\"https://static.igem.wiki/teams/4160/wiki/footer/wit-logo-officieel.svg\"/>\n",
      "      <div class=\"sociallogos\">\n",
      "       <a href=\"mailto:igem@tue.nl\">\n",
      "        <img class=\"social-icon\" src=\"https://static.igem.wiki/teams/4160/wiki/footer/mail.svg\"/>\n",
      "       </a>\n",
      "       <a href=\"https://www.linkedin.com/company/igem-tu-eindhoven/mycompany/\" target=\"_blank\">\n",
      "        <img class=\"social-icon\" src=\"https://static.igem.wiki/teams/4160/wiki/footer/linkedin.svg\"/>\n",
      "       </a>\n",
      "       <a href=\"https://www.facebook.com/iGEMTUEindhoven/\" target=\"_blank\">\n",
      "        <img class=\"social-icon\" src=\"https://static.igem.wiki/teams/4160/wiki/footer/facebook.svg\"/>\n",
      "       </a>\n",
      "       <a href=\"https://www.instagram.com/igem_eindhoven/\" target=\"_blank\">\n",
      "        <img class=\"social-icon\" src=\"https://static.igem.wiki/teams/4160/wiki/footer/insta.svg\"/>\n",
      "       </a>\n",
      "       <a href=\"https://twitter.com/eindhoven_igem\" target=\"_blank\">\n",
      "        <img class=\"social-icon\" src=\"https://static.igem.wiki/teams/4160/wiki/footer/twitter.svg\"/>\n",
      "       </a>\n",
      "       <!--\n",
      "          <center><img src=\"https://static.igem.wiki/teams/4160/wiki/footer/socials.svg\" alt=\"socials\" style=\"width:20%\"></center>\n",
      "            -->\n",
      "      </div>\n",
      "     </div>\n",
      "     <div class=\"w-100 d-lg-none d-block\">\n",
      "     </div>\n",
      "     <div class=\"col-lg-2 col-sm-4 col 6 d-flex flex-column align-items-center\">\n",
      "      <h4 class=\"mt-lg-0 mt-sm-4 mb-3\">\n",
      "       Contact\n",
      "      </h4>\n",
      "      <p>\n",
      "       <a href=\"mailto:igem@tue.nl\">\n",
      "        igem@tue.nl\n",
      "       </a>\n",
      "      </p>\n",
      "     </div>\n",
      "     <div class=\"col-lg-2 col-sm-4 col 6 d-flex flex-column align-items-center\">\n",
      "      <div class=\"col-lg-3 col-xs-12\">\n",
      "       <h4 class=\"mt-lg-0 mt-sm-3\">\n",
      "        Follow\n",
      "       </h4>\n",
      "       <ul class=\"m-2 p-2\">\n",
      "        <li>\n",
      "         <a href=\"https://www.linkedin.com/company/igem-tu-eindhoven/mycompany/\" target=\"_blank\">\n",
      "          LinkedIn\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a href=\"https://twitter.com/eindhoven_igem\" target=\"_blank\">\n",
      "          Twitter\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a href=\"https://www.instagram.com/igem_eindhoven/\" target=\"_blank\">\n",
      "          Instagram\n",
      "         </a>\n",
      "        </li>\n",
      "        <li>\n",
      "         <a href=\"https://www.facebook.com/iGEMTUEindhoven/\" target=\"_blank\">\n",
      "          Facebook\n",
      "         </a>\n",
      "        </li>\n",
      "       </ul>\n",
      "      </div>\n",
      "     </div>\n",
      "     <!-- The following MUST be on every page: license information and link to the repository on gitlab.igem.org -->\n",
      "     <div class=\"row mt-4\">\n",
      "      <div class=\"col d-flex flex-column align-items-center\">\n",
      "       <p class=\"mb-0\">\n",
      "        <small>\n",
      "         Â© 2022 - Content on this site is licensed under a\n",
      "         <a class=\"subfoot\" href=\"https://creativecommons.org/licenses/by/4.0/\" rel=\"license\">\n",
      "          Creative Commons Attribution 4.0\n",
      "                International license\n",
      "         </a>\n",
      "         .\n",
      "        </small>\n",
      "       </p>\n",
      "       <p>\n",
      "        <small>\n",
      "         The repository used to create this website is available at\n",
      "         <a href=\"https://gitlab.igem.org/2022/tu-eindhoven\">\n",
      "          gitlab.igem.org/2022/tu-eindhoven\n",
      "         </a>\n",
      "         .\n",
      "        </small>\n",
      "       </p>\n",
      "      </div>\n",
      "     </div>\n",
      "    </div>\n",
      "   </div>\n",
      "  </footer>\n",
      "  <!-- Wiki Tools: Teams are allowed to remove it -->\n",
      "  <style>\n",
      "   .wiki-tools-footer {\n",
      "        --bs-gutter-x: 1.5rem;\n",
      "        background: #222124;\n",
      "        margin: 0;\n",
      "        max-width: 100%;\n",
      "        padding: 1rem 3rem;\n",
      "        display: flex;\n",
      "        gap: 1.5em;\n",
      "    }\n",
      "    .wiki-tools-footer a {\n",
      "        text-decoration: none;\n",
      "        color: white;\n",
      "        font-weight: 400;\n",
      "        transition: color 0.3s;\n",
      "    }\n",
      "    .wiki-tools-footer a:hover {\n",
      "        color: #08C784;\n",
      "    }\n",
      "    .wiki-tools-footer img {\n",
      "        object-fit: contain;\n",
      "    }\n",
      "  </style>\n",
      "  <section class=\"wiki-tools-footer container\">\n",
      "   <a id=\"wiki-tools-link\">\n",
      "    Access Wiki Tools\n",
      "   </a>\n",
      "   <a id=\"wiki-tools-repo\">\n",
      "    View Source Code\n",
      "   </a>\n",
      "   <img id=\"wiki-tools-pipeline\"/>\n",
      "  </section>\n",
      "  <script>\n",
      "   const wikiToolsLink = document.getElementById('wiki-tools-link')\n",
      "    const wikiToolsRepo = document.getElementById('wiki-tools-repo')\n",
      "    const wikiToolsPipeline = document.getElementById('wiki-tools-pipeline')\n",
      "\n",
      "    const path = document.location.pathname.split('/')\n",
      "\n",
      "    const competitionSlug = document.location.host.split('.')[0]\n",
      "    const teamSlug = path[1]\n",
      "    const page = path[2] === '' ? 'index' : path[2]\n",
      "\n",
      "    wikiToolsLink.href = `https://tools.igem.org/wiki-tools/${competitionSlug}/${teamSlug}/${page}`\n",
      "    wikiToolsRepo.href = `https://gitlab.igem.org/${competitionSlug}/${teamSlug}`\n",
      "    wikiToolsPipeline.src = `https://gitlab.igem.org/${competitionSlug}/${teamSlug}/badges/main/pipeline.svg`\n",
      "  </script>\n",
      "  <!-- Bootstrap Bundle with Popper -->\n",
      "  <script src=\"static/bootstrap.bundle.min.js\">\n",
      "  </script>\n",
      " </body>\n",
      "</html>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(soup.prettify())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "virtualenv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
